Index: tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- tools/clang/lib/Driver/ToolChains.cpp	(révision 144213)
+++ tools/clang/lib/Driver/ToolChains.cpp	(copie de travail)
@@ -1818,6 +1818,11 @@
   if (IsRedhat(Distro))
     ExtraOpts.push_back("--no-add-needed");
 
+  /* See: http://wiki.debian.org/ToolChain/DSOLinking */
+  if (IsDebian(Distro))
+    ExtraOpts.push_back("--no-copy-dt-needed-entries");
+
+
   if (Distro == DebianSqueeze || Distro == DebianWheezy ||
       IsOpenSuse(Distro) ||
       (IsRedhat(Distro) && Distro != RHEL4 && Distro != RHEL5) ||
