Index: clang-3.0/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.0.orig/tools/clang/lib/Driver/ToolChains.cpp	2011-12-07 18:09:23.000000000 +0100
+++ clang-3.0/tools/clang/lib/Driver/ToolChains.cpp	2011-12-07 18:10:41.000000000 +0100
@@ -1484,9 +1484,9 @@
     StringRef Data = File.get()->getBuffer();
     if (Data[0] == '5')
       return DebianLenny;
-    else if (Data.startswith("squeeze/sid"))
+    else if (Data.startswith("squeeze/sid") || Data[0] == '6')
       return DebianSqueeze;
-    else if (Data.startswith("wheezy/sid"))
+    else if (Data.startswith("wheezy/sid")  || Data[0] == '7')
       return DebianWheezy;
     return UnknownDistro;
   }
