Index: llvm-toolchain-snapshot_3.4~svn193603/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193603.orig/lldb/lib/Makefile	2013-10-29 12:26:27.542137766 +0100
+++ llvm-toolchain-snapshot_3.4~svn193603/lldb/lib/Makefile	2013-10-29 12:26:27.542137766 +0100
@@ -158,5 +158,6 @@
     # Allow unresolved symbols.
     LLVMLibsOptions += -Wl,--allow-shlib-undefined
     # Link in python
-    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -L/usr/local/lib -lexecinfo
+    LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt
+# -L/usr/local/lib -lexecinfo
 endif
