---
 llvm-ld.cpp |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

--- llvm-2.9-2.9+dfsg.orig/tools/llvm-ld/llvm-ld.cpp	2011-02-09 05:17:39.000000000 +0100
+++ llvm-2.9-2.9+dfsg/tools/llvm-ld/llvm-ld.cpp	2012-03-10 17:57:17.113540243 +0100
@@ -434,7 +434,17 @@
 
   Out2.os() << "#!/bin/sh\n";
   // Allow user to setenv LLVMINTERP if lli is not in their PATH.
-  Out2.os() << "lli=${LLVMINTERP-lli}\n";
+  Out2.os() << "lli=${LLVMINTERP-lli-" << PACKAGE_VERSION << "}\n";
+  Out2.os() << "command -v $lli >/dev/null 2>&1\n";
+  Out2.os() << "if test $? -ne 0; then \n";
+  Out2.os() << "    lli=${LLVMINTERP-lli}\n";
+  Out2.os() << "    command -v $lli >/dev/null 2>&1\n";
+  Out2.os() << "    if test $? -ne 0; then \n";
+  Out2.os() << "        echo 'Could not find either lli-" << PACKAGE_VERSION << " or lli'\n";
+  Out2.os() << "        exit 42\n";
+  Out2.os() << "    fi\n";
+  Out2.os() << "fi\n";
+
   Out2.os() << "exec $lli \\\n";
   // gcc accepts -l<lib> and implicitly searches /lib and /usr/lib.
   LibPaths.push_back("/lib");
