Index: llvm-toolchain-3.3-3.3~0rc2/include/llvm/Support/Host.h
===================================================================
--- llvm-toolchain-3.3-3.3~0rc2.orig/include/llvm/Support/Host.h	2013-04-15 22:13:59.000000000 +0200
+++ llvm-toolchain-3.3-3.3~0rc2/include/llvm/Support/Host.h	2013-06-05 11:13:28.000000000 +0200
@@ -16,7 +16,7 @@
 
 #include "llvm/ADT/StringMap.h"
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__GNU__)
 #include <endian.h>
 #else
 #ifndef LLVM_ON_WIN32
