Index: llvm/lib/ExecutionEngine/JIT/JIT.cpp
===================================================================
--- llvm.orig/lib/ExecutionEngine/JIT/JIT.cpp	2013-06-03 17:32:40.000000000 +0200
+++ llvm/lib/ExecutionEngine/JIT/JIT.cpp	2013-06-03 17:32:59.000000000 +0200
@@ -68,7 +68,7 @@
 }
 
 // Determine whether we can register EH tables.
-#if (defined(__GNUC__) && !defined(__ARM_EABI__) && \
+#if (defined(__GNUC__) && !defined(__ARM_EABI__) && !defined(__ia64__) && \
      !defined(__USING_SJLJ_EXCEPTIONS__))
 #define HAVE_EHTABLE_SUPPORT 1
 #else
