diff -urN debian/changelog llvm-toolchain-snapshot_3.3~svn183554/debian/changelog
--- debian/changelog	2013-12-18 19:33:46.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/changelog	2014-01-11 21:59:12.271133052 +0900
@@ -1,3 +1,9 @@
+llvm-toolchain-3.3 (1:3.3-debian16ppa3) precise; urgency=low
+
+  * ppa for precise
+
+ -- ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>  Sat, 11 Jan 2014 21:45:00 +0900
+
 llvm-toolchain-3.3 (1:3.3-16) unstable; urgency=low
 
   * Make lldb-3.3-dev dependencies matches lldb-3.3.
diff -urN debian/clang-3.3-relaxng.install llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3-relaxng.install
--- debian/clang-3.3-relaxng.install	1970-01-01 09:00:00.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3-relaxng.install	2013-09-04 18:21:38.835126651 +0900
@@ -0,0 +1 @@
+clang/bindings/xml/comment-xml-schema.rng	/usr/share/xml/clang-3.3-relaxng/
diff -urN debian/clang-3.3.install llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.install
--- debian/clang-3.3.install	2013-11-05 21:23:55.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.install	2013-11-18 16:21:48.421986005 +0900
@@ -1,10 +1,9 @@
-usr/lib/llvm-3.3/bin/clang			/usr/bin/
-usr/lib/llvm-3.3/bin/clang++			/usr/bin/
-usr/lib/llvm-3.3/bin/clang-check			/usr/bin/
-usr/lib/llvm-3.3/bin/clang-tblgen			/usr/bin/
-usr/lib/llvm-3.3/bin/c-index-test*
-usr/bin/c-index-test*
-tools/clang/tools/scan-build			/usr/share/clang/
-tools/clang/tools/scan-view			/usr/share/clang/
+usr/lib/llvm-3.3/bin/clang		/usr/lib/llvm-3.3/bin/
+usr/lib/llvm-3.3/bin/clang++		/usr/lib/llvm-3.3/bin/
+usr/lib/llvm-3.3/bin/clang-check	/usr/lib/llvm-3.3/bin/
+usr/lib/llvm-3.3/bin/clang-tblgen	/usr/lib/llvm-3.3/bin/
+usr/lib/llvm-3.3/bin/c-index-test	/usr/lib/llvm-3.3/bin/
+tools/clang/tools/scan-build		/usr/lib/llvm-3.3/share/
+tools/clang/tools/scan-view		/usr/lib/llvm-3.3/share/
 #usr/share/man/man1/clang.1	usr/share/man/man1/
-debian/pollycc		/usr/bin/
+debian/pollycc				/usr/lib/llvm-3.3/bin/
diff -urN debian/clang-3.3.links llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.links
--- debian/clang-3.3.links	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.links	2013-11-18 16:22:59.518338555 +0900
@@ -1,5 +1,12 @@
-usr/share/man/man1/clang-3.3.1.gz		usr/share/man/man1/llvm-clang.1.gz
-usr/share/man/man1/clang-3.3.1.gz		usr/share/man/man1/clang++.1.gz
-/usr/share/clang/scan-build/scan-build	/usr/bin/scan-build
-/usr/share/clang/scan-view/scan-view	/usr/bin/scan-view
-/usr/bin/clang				/usr/bin/llvm-clang
+/usr/lib/llvm-3.3/bin/clang			/usr/bin/llvm-clang-3.3
+/usr/lib/llvm-3.3/bin/clang			/usr/bin/clang-3.3
+/usr/lib/llvm-3.3/bin/clang++			/usr/bin/clang++-3.3
+/usr/lib/llvm-3.3/bin/clang-check		/usr/bin/clang-check-3.3
+/usr/lib/llvm-3.3/bin/clang-tblgen		/usr/bin/clang-tblgen-3.3
+/usr/lib/llvm-3.3/bin/c-index-test		/usr/bin/c-index-test-3.3
+/usr/lib/llvm-3.3/share/scan-build/scan-build	/usr/bin/scan-build-3.3
+/usr/lib/llvm-3.3/share/scan-view/scan-view	/usr/bin/scan-view-3.3
+/usr/lib/llvm-3.3/bin/pollycc			/usr/bin/pollycc-3.3
+/usr/lib/llvm-3.3/bin/asan_symbolize		/usr/bin/asan_symbolize-3.3
+usr/share/man/man1/clang-3.3.1.gz		usr/share/man/man1/llvm-clang-3.3.1.gz
+usr/share/man/man1/clang-3.3.1.gz		usr/share/man/man1/clang++-3.3.1.gz
diff -urN debian/clang-3.3.manpages llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.manpages
--- debian/clang-3.3.manpages	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.manpages	2013-09-04 18:21:38.835126651 +0900
@@ -1,5 +1,5 @@
 build-llvm/tools/clang/docs/tools/clang-3.3.1
-clang/tools/scan-build/scan-build.1
-debian/man/clang-check.1
-debian/man/clang-tblgen.1
-debian/man/scan-view.1
+debian/man/scan-build-3.3.1
+debian/man/clang-check-3.3.1
+debian/man/clang-tblgen-3.3.1
+debian/man/scan-view-3.3.1
diff -urN debian/clang-3.3.postinst llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.postinst
--- debian/clang-3.3.postinst	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.postinst	1970-01-01 09:00:00.000000000 +0900
@@ -1,31 +0,0 @@
-#! /bin/sh -e
-
-prio=10
-
-update-alternatives --quiet \
-	--install /usr/bin/c++ \
-		c++ \
-		/usr/bin/clang++ \
-		$prio
-
-update-alternatives --quiet \
-	--install /usr/bin/cc \
-		cc \
-		/usr/bin/clang \
-		$prio
-
-update-alternatives --quiet \
-	--install /usr/bin/c89 \
-		c89 \
-		/usr/bin/clang \
-		$prio
-
-update-alternatives --quiet \
-	--install /usr/bin/c99 \
-		c99 \
-		/usr/bin/clang \
-		$prio
-
-#DEBHELPER#
-
-exit 0
diff -urN debian/clang-3.3.prerm llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.prerm
--- debian/clang-3.3.prerm	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-3.3.prerm	1970-01-01 09:00:00.000000000 +0900
@@ -1,14 +0,0 @@
-#!/bin/sh
-
-set -e
-
-if [ "$1" = "remove" ] || [ "$1" = "deconfigure" ]; then
-    update-alternatives --quiet --remove cc /usr/bin/clang
-    update-alternatives --quiet --remove c89 /usr/bin/clang
-    update-alternatives --quiet --remove c99 /usr/bin/clang
-    update-alternatives --quiet --remove c++ /usr/bin/clang++
-fi
-
-#DEBHELPER#
-
-exit 0
diff -urN debian/clang-format-3.3.install llvm-toolchain-snapshot_3.3~svn183554/debian/clang-format-3.3.install
--- debian/clang-format-3.3.install	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-format-3.3.install	2013-09-04 18:21:38.835126651 +0900
@@ -1,5 +1,3 @@
-clang/tools/clang-format/clang-format-3.3.py usr/share/vim/addons/syntax/
-clang/tools/clang-format/clang-format-diff-3.3 /usr/bin/
+clang/tools/clang-format/clang-format.py	/usr/lib/llvm-3.3/bin/
+clang/tools/clang-format/clang-format-diff.py	/usr/lib/llvm-3.3/bin/
 clang/tools/clang-format/clang-format.el usr/share/emacs/site-lisp/clang-format-3.3/
-usr/bin/clang-format-3.3
-usr/lib/llvm-3.3/bin/clang-format
diff -urN debian/clang-format-3.3.links llvm-toolchain-snapshot_3.3~svn183554/debian/clang-format-3.3.links
--- debian/clang-format-3.3.links	1970-01-01 09:00:00.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/clang-format-3.3.links	2013-09-04 18:21:38.835126651 +0900
@@ -0,0 +1,3 @@
+/usr/lib/llvm-3.3/bin/clang-format.py		/usr/share/vim/addons/syntax/clang-format-3.3.py
+/usr/lib/llvm-3.3/bin/clang-format.py		/usr/bin/clang-format-3.3
+/usr/lib/llvm-3.3/bin/clang-format-diff.py	/usr/bin/clang-format-diff-3.3
diff -urN debian/control llvm-toolchain-snapshot_3.3~svn183554/debian/control
--- debian/control	2013-12-18 19:33:46.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/control	2013-12-12 02:05:38.246043381 +0900
@@ -1,20 +1,18 @@
 Source: llvm-toolchain-3.3
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 9.0), flex, bison, dejagnu, tcl, expect,
     autoconf, automake, perl, libtool, doxygen, chrpath, texinfo,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
     lsb-release, patchutils, diffstat, xz-utils, chrpath, python-dev,
     libedit-dev, swig, python-sphinx, ocaml-nox, binutils-dev, libcloog-isl-dev,
     libisl-dev (>= 0.11.1), libjsoncpp-dev,
-    lcov, procps, help2man, dh-ocaml
+    lcov, procps, help2man
 Build-Conflicts: oprofile, ocaml
 Standards-Version: 3.9.5
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/llvm-toolchain/branches/3.3/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/llvm-toolchain/branches/3.3/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 # ------------- clang -------------
 
@@ -25,10 +23,6 @@
  libclang1-3.3 (= ${binary:Version})
 Provides: c-compiler, objc-compiler, c++-compiler
 Recommends: llvm-3.3-dev, python
-Replaces: clang (<< 1:3.3), clang-3.1, clang-3.2, clang-3.4, compiler-rt,
- llvm-3.3 (<< 1:3.3-13)
-Breaks: clang (<< 1:3.3), clang-3.1, clang-3.2, clang-3.4, compiler-rt,
- llvm-3.3 (<< 1:3.3-13)
 Description: C, C++ and Objective-C compiler (LLVM based)
  Clang project is a C, C++, Objective C and Objective C++ front-end
  for the LLVM compiler. Its goal is to offer a replacement to the GNU Compiler
@@ -40,8 +34,6 @@
 Package: clang-format-3.3
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, python
-Replaces: clang-format-3.4
-Breaks: clang-format-3.4
 Description: Tool to format C/C++/Obj-C code
  Clang-format is both a library and a stand-alone tool with the goal of
  automatically reformatting C++ sources files according to configurable
@@ -57,8 +49,6 @@
 Package: cpp11-migrate-3.3
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}
-Replaces: cpp11-migrate-3.4
-Breaks: cpp11-migrate-3.4
 Description: Tool to convert C++98 and C++03 code to C++11
  The purpose of the C++11 Migrator is to do source-to-source translation to
  migrate existing C++ code to use C++11 features to enhance
@@ -102,12 +92,9 @@
  locations with elements within the AST, and other facilities that support
  Clang-based development tools.
 
-
-Package: libclang1-3.3-dbg
-Architecture: any
-Section: debug
-Priority: extra
-Depends: ${shlibs:Depends}, ${misc:Depends}, ${dep:devlibs}, libclang1-3.3
+Package: clang-3.3-relaxng
+Architecture: all
+Section: misc
 Description: clang library
  Clang project is a C, C++, Objective C and Objective C++ front-end
  for the LLVM compiler. Its goal is to offer a replacement to the GNU Compiler
@@ -116,7 +103,7 @@
  Clang implements all of the ISO C++ 1998 and 2001 standards and also provides
  a partial support of C++1y.
  .
- This package contains the debugging symbols.
+ This package contains the clang Relax NG schema for the XML.
 
 Package: libclang-3.3-dev
 Architecture: any
@@ -138,8 +125,7 @@
 Architecture: any
 Section: libdevel
 Depends: ${shlibs:Depends}, ${misc:Depends}, libllvm3.3 (= ${binary:Version})
-Replaces: llvm-3.3-dev (<< 1:3.2repack), libclang-3.3-dev
-Breaks: llvm-3.3-dev (<< 1:3.2repack)
+Replaces: libclang-3.3-dev
 Description: clang library - Common development package
  Clang project is a C, C++, Objective C and Objective C++ front-end
  for the LLVM compiler. Its goal is to offer a replacement to the GNU Compiler
@@ -157,8 +143,6 @@
 Section: python
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, python
-Replaces: python-clang-3.4
-Breaks: python-clang-3.4
 Description: Clang Python Bindings
  This binding package provides access to the Clang compiler and libraries.
 
@@ -193,27 +177,11 @@
  This package contains the LLVM runtime library.
 
 
-Package: libllvm3.3-dbg
-Architecture: any
-Section: debug
-Priority: extra
-Depends: ${shlibs:Depends}, ${misc:Depends}, libllvm3.3
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: Modular compiler and toolchain technologies, debugging libraries
- LLVM is a collection of libraries and tools that make it easy to build
- compilers, optimizers, just-in-time code generators, and many other
- compiler-related programs.
- .
- This package contains the LLVM runtime library debug symbols.
-
 Package: llvm-3.3
 Architecture: any
 Suggests: llvm-3.3-doc
 Depends: llvm-3.3-runtime (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
 Recommends: llvm-3.3-dev
-Conflicts: llvm (<< 2.7-1)
-Replaces: llvm (<< 2.7-1)
 Description: Modular compiler and toolchain technologies
  LLVM is a collection of libraries and tools that make it easy to build
  compilers, optimizers, just-in-time code generators, and many other
@@ -243,8 +211,6 @@
 Package: llvm-3.3-runtime
 Architecture: any
 Depends: binfmt-support, ${shlibs:Depends}, ${misc:Depends}
-Conflicts: llvm (<< 2.7-1)
-Replaces: llvm (<< 2.7-1)
 Description: Modular compiler and toolchain technologies, IR interpreter
  LLVM is a collection of libraries and tools that make it easy to build
  compilers, optimizers, just-in-time code generators, and many other
@@ -264,7 +230,6 @@
 Package: llvm-3.3-dev
 Architecture: any
 Depends: ${shlibs:Depends}, libffi-dev (>= 3.0.9), ${misc:Depends}, llvm-3.3 (= ${binary:Version})
-Replaces: llvm (<< 2.2-3)
 Description: Modular compiler and toolchain technologies, libraries and headers
  LLVM is a collection of libraries and tools that make it easy to build
  compilers, optimizers, just-in-time code generators, and many other
@@ -342,29 +307,3 @@
  .
  This package contains examples for using LLVM, both in developing
  extensions to LLVM and in using it to compile code.
-
-# ------------- lldb -------------
-
-Package: lldb-3.3
-Architecture: amd64 armel armhf i386  kfreebsd-amd64 kfreebsd-i386 s390 s390x sparc alpha arm64 hppa m68k powerpcspe ppc64 sh4 sparc64 x32 powerpc
-# ia64 mips mipsel hurd have been removed
-Depends: ${shlibs:Depends}, ${misc:Depends}, libllvm3.3 (= ${binary:Version}), python, llvm-3.3-dev
-Pre-Depends: ${misc:Pre-Depends}
-Replaces: lldb-3.2
-Breaks: lldb-3.2
-Description: Next generation, high-performance debugger
- LLDB is a next generation, high-performance debugger. It is built as a set of
- reusable components which highly leverage existing libraries in the larger LLVM
- Project, such as the Clang expression parser and LLVM disassembler.
-
-Package: lldb-3.3-dev
-Architecture: amd64 armel armhf i386  kfreebsd-amd64 kfreebsd-i386 s390 s390x sparc alpha arm64 hppa m68k powerpcspe ppc64 sh4 sparc64 x32 powerpc
-# ia64 mips mipsel hurd have been removed
-Depends: ${shlibs:Depends}, ${misc:Depends}, lldb-3.3 (= ${binary:Version})
-Pre-Depends: ${misc:Pre-Depends}
-Description: Next generation, high-performance debugger - Header files
- LLDB is a next generation, high-performance debugger. It is built as a set of
- reusable components which highly leverage existing libraries in the larger LLVM
- Project, such as the Clang expression parser and LLVM disassembler.
- .
- This package provides the header files to build extension over lldb.
diff -urN debian/libclang1-3.3.symbols llvm-toolchain-snapshot_3.3~svn183554/debian/libclang1-3.3.symbols
--- debian/libclang1-3.3.symbols	2013-10-16 16:21:51.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/libclang1-3.3.symbols	2013-10-25 12:29:59.708295647 +0900
@@ -1,277 +1,277 @@
 libclang.so.1 libclang1-3.3 #MINVER#
- clang_BlockCommandComment_getArgText@Base 1:3.3
- clang_BlockCommandComment_getCommandName@Base 1:3.3
- clang_BlockCommandComment_getNumArgs@Base 1:3.3
- clang_BlockCommandComment_getParagraph@Base 1:3.3
- clang_CXCursorSet_contains@Base 1:3.3
- clang_CXCursorSet_insert@Base 1:3.3
- clang_CXIndex_getGlobalOptions@Base 1:3.3
- clang_CXIndex_setGlobalOptions@Base 1:3.3
- clang_CXXMethod_isStatic@Base 1:3.3
- clang_CXXMethod_isVirtual@Base 1:3.3
- clang_Comment_getChild@Base 1:3.3
- clang_Comment_getKind@Base 1:3.3
- clang_Comment_getNumChildren@Base 1:3.3
- clang_Comment_isWhitespace@Base 1:3.3
- clang_CompilationDatabase_dispose@Base 1:3.3
- clang_CompilationDatabase_fromDirectory@Base 1:3.3
- clang_CompilationDatabase_getAllCompileCommands@Base 1:3.3
- clang_CompilationDatabase_getCompileCommands@Base 1:3.3
- clang_CompileCommand_getArg@Base 1:3.3
- clang_CompileCommand_getDirectory@Base 1:3.3
- clang_CompileCommand_getNumArgs@Base 1:3.3
- clang_CompileCommands_dispose@Base 1:3.3
- clang_CompileCommands_getCommand@Base 1:3.3
- clang_CompileCommands_getSize@Base 1:3.3
- clang_Cursor_getArgument@Base 1:3.3
- clang_Cursor_getBriefCommentText@Base 1:3.3
- clang_Cursor_getCommentRange@Base 1:3.3
- clang_Cursor_getModule@Base 1:3.3
- clang_Cursor_getNumArguments@Base 1:3.3
- clang_Cursor_getObjCDeclQualifiers@Base 1:3.3
- clang_Cursor_getObjCPropertyAttributes@Base 1:3.3
- clang_Cursor_getObjCSelectorIndex@Base 1:3.3
- clang_Cursor_getParsedComment@Base 1:3.3
- clang_Cursor_getRawCommentText@Base 1:3.3
- clang_Cursor_getReceiverType@Base 1:3.3
- clang_Cursor_getSpellingNameRange@Base 1:3.3
- clang_Cursor_getTranslationUnit@Base 1:3.3
- clang_Cursor_isBitField@Base 1:3.3
- clang_Cursor_isDynamicCall@Base 1:3.3
- clang_Cursor_isNull@Base 1:3.3
- clang_Cursor_isVariadic@Base 1:3.3
- clang_FullComment_getAsHTML@Base 1:3.3
- clang_FullComment_getAsXML@Base 1:3.3
- clang_HTMLStartTagComment_isSelfClosing@Base 1:3.3
- clang_HTMLStartTag_getAttrName@Base 1:3.3
- clang_HTMLStartTag_getAttrValue@Base 1:3.3
- clang_HTMLStartTag_getNumAttrs@Base 1:3.3
- clang_HTMLTagComment_getAsString@Base 1:3.3
- clang_HTMLTagComment_getTagName@Base 1:3.3
- clang_IndexAction_create@Base 1:3.3
- clang_IndexAction_dispose@Base 1:3.3
- clang_InlineCommandComment_getArgText@Base 1:3.3
- clang_InlineCommandComment_getCommandName@Base 1:3.3
- clang_InlineCommandComment_getNumArgs@Base 1:3.3
- clang_InlineCommandComment_getRenderKind@Base 1:3.3
- clang_InlineContentComment_hasTrailingNewline@Base 1:3.3
- clang_Location_isInSystemHeader@Base 1:3.3
- clang_Module_getASTFile@Base 1:3.3
- clang_Module_getFullName@Base 1:3.3
- clang_Module_getName@Base 1:3.3
- clang_Module_getNumTopLevelHeaders@Base 1:3.3
- clang_Module_getParent@Base 1:3.3
- clang_Module_getTopLevelHeader@Base 1:3.3
- clang_ParamCommandComment_getDirection@Base 1:3.3
- clang_ParamCommandComment_getParamIndex@Base 1:3.3
- clang_ParamCommandComment_getParamName@Base 1:3.3
- clang_ParamCommandComment_isDirectionExplicit@Base 1:3.3
- clang_ParamCommandComment_isParamIndexValid@Base 1:3.3
- clang_Range_isNull@Base 1:3.3
- clang_TParamCommandComment_getDepth@Base 1:3.3
- clang_TParamCommandComment_getIndex@Base 1:3.3
- clang_TParamCommandComment_getParamName@Base 1:3.3
- clang_TParamCommandComment_isParamPositionValid@Base 1:3.3
- clang_TextComment_getText@Base 1:3.3
- clang_Type_getAlignOf@Base 1:3.3
- clang_Type_getOffsetOf@Base 1:3.3
- clang_Type_getSizeOf@Base 1:3.3
- clang_VerbatimBlockLineComment_getText@Base 1:3.3
- clang_VerbatimLineComment_getText@Base 1:3.3
- clang_annotateTokens@Base 1:3.3
- clang_codeCompleteAt@Base 1:3.3
- clang_codeCompleteGetContainerKind@Base 1:3.3
- clang_codeCompleteGetContainerUSR@Base 1:3.3
- clang_codeCompleteGetContexts@Base 1:3.3
- clang_codeCompleteGetDiagnostic@Base 1:3.3
- clang_codeCompleteGetNumDiagnostics@Base 1:3.3
- clang_codeCompleteGetObjCSelector@Base 1:3.3
- clang_constructUSR_ObjCCategory@Base 1:3.3
- clang_constructUSR_ObjCClass@Base 1:3.3
- clang_constructUSR_ObjCIvar@Base 1:3.3
- clang_constructUSR_ObjCMethod@Base 1:3.3
- clang_constructUSR_ObjCProperty@Base 1:3.3
- clang_constructUSR_ObjCProtocol@Base 1:3.3
- clang_createCXCursorSet@Base 1:3.3
- clang_createIndex@Base 1:3.3
- clang_createTranslationUnit@Base 1:3.3
- clang_createTranslationUnitFromSourceFile@Base 1:3.3
- clang_defaultCodeCompleteOptions@Base 1:3.3
- clang_defaultDiagnosticDisplayOptions@Base 1:3.3
- clang_defaultEditingTranslationUnitOptions@Base 1:3.3
- clang_defaultReparseOptions@Base 1:3.3
- clang_defaultSaveOptions@Base 1:3.3
- clang_disposeCXCursorSet@Base 1:3.3
- clang_disposeCXPlatformAvailability@Base 1:3.3
- clang_disposeCXTUResourceUsage@Base 1:3.3
- clang_disposeCodeCompleteResults@Base 1:3.3
- clang_disposeDiagnostic@Base 1:3.3
- clang_disposeDiagnosticSet@Base 1:3.3
- clang_disposeIndex@Base 1:3.3
- clang_disposeOverriddenCursors@Base 1:3.3
- clang_disposeString@Base 1:3.3
- clang_disposeTokens@Base 1:3.3
- clang_disposeTranslationUnit@Base 1:3.3
- clang_enableStackTraces@Base 1:3.3
- clang_equalCursors@Base 1:3.3
- clang_equalLocations@Base 1:3.3
- clang_equalRanges@Base 1:3.3
- clang_equalTypes@Base 1:3.3
- clang_executeOnThread@Base 1:3.3
- clang_findIncludesInFile@Base 1:3.3
- clang_findIncludesInFileWithBlock@Base 1:3.3
- clang_findReferencesInFile@Base 1:3.3
- clang_findReferencesInFileWithBlock@Base 1:3.3
- clang_formatDiagnostic@Base 1:3.3
- clang_getArgType@Base 1:3.3
- clang_getArrayElementType@Base 1:3.3
- clang_getArraySize@Base 1:3.3
- clang_getCString@Base 1:3.3
- clang_getCXTUResourceUsage@Base 1:3.3
- clang_getCXXAccessSpecifier@Base 1:3.3
- clang_getCanonicalCursor@Base 1:3.3
- clang_getCanonicalType@Base 1:3.3
- clang_getChildDiagnostics@Base 1:3.3
- clang_getClangVersion@Base 1:3.3
- clang_getCompletionAnnotation@Base 1:3.3
- clang_getCompletionAvailability@Base 1:3.3
- clang_getCompletionBriefComment@Base 1:3.3
- clang_getCompletionChunkCompletionString@Base 1:3.3
- clang_getCompletionChunkKind@Base 1:3.3
- clang_getCompletionChunkText@Base 1:3.3
- clang_getCompletionNumAnnotations@Base 1:3.3
- clang_getCompletionParent@Base 1:3.3
- clang_getCompletionPriority@Base 1:3.3
- clang_getCursor@Base 1:3.3
- clang_getCursorAvailability@Base 1:3.3
- clang_getCursorCompletionString@Base 1:3.3
- clang_getCursorDefinition@Base 1:3.3
- clang_getCursorDisplayName@Base 1:3.3
- clang_getCursorExtent@Base 1:3.3
- clang_getCursorKind@Base 1:3.3
- clang_getCursorKindSpelling@Base 1:3.3
- clang_getCursorLanguage@Base 1:3.3
- clang_getCursorLexicalParent@Base 1:3.3
- clang_getCursorLinkage@Base 1:3.3
- clang_getCursorLocation@Base 1:3.3
- clang_getCursorPlatformAvailability@Base 1:3.3
- clang_getCursorReferenceNameRange@Base 1:3.3
- clang_getCursorReferenced@Base 1:3.3
- clang_getCursorResultType@Base 1:3.3
- clang_getCursorSemanticParent@Base 1:3.3
- clang_getCursorSpelling@Base 1:3.3
- clang_getCursorType@Base 1:3.3
- clang_getCursorUSR@Base 1:3.3
- clang_getDeclObjCTypeEncoding@Base 1:3.3
- clang_getDefinitionSpellingAndExtent@Base 1:3.3
- clang_getDiagnostic@Base 1:3.3
- clang_getDiagnosticCategory@Base 1:3.3
- clang_getDiagnosticCategoryName@Base 1:3.3
- clang_getDiagnosticCategoryText@Base 1:3.3
- clang_getDiagnosticFixIt@Base 1:3.3
- clang_getDiagnosticInSet@Base 1:3.3
- clang_getDiagnosticLocation@Base 1:3.3
- clang_getDiagnosticNumFixIts@Base 1:3.3
- clang_getDiagnosticNumRanges@Base 1:3.3
- clang_getDiagnosticOption@Base 1:3.3
- clang_getDiagnosticRange@Base 1:3.3
- clang_getDiagnosticSetFromTU@Base 1:3.3
- clang_getDiagnosticSeverity@Base 1:3.3
- clang_getDiagnosticSpelling@Base 1:3.3
- clang_getElementType@Base 1:3.3
- clang_getEnumConstantDeclUnsignedValue@Base 1:3.3
- clang_getEnumConstantDeclValue@Base 1:3.3
- clang_getEnumDeclIntegerType@Base 1:3.3
- clang_getExpansionLocation@Base 1:3.3
- clang_getFieldDeclBitWidth@Base 1:3.3
- clang_getFile@Base 1:3.3
- clang_getFileLocation@Base 1:3.3
- clang_getFileName@Base 1:3.3
- clang_getFileTime@Base 1:3.3
- clang_getFileUniqueID@Base 1:3.3
- clang_getFunctionTypeCallingConv@Base 1:3.3
- clang_getIBOutletCollectionType@Base 1:3.3
- clang_getIncludedFile@Base 1:3.3
- clang_getInclusions@Base 1:3.3
- clang_getInstantiationLocation@Base 1:3.3
- clang_getLocation@Base 1:3.3
- clang_getLocationForOffset@Base 1:3.3
- clang_getNullCursor@Base 1:3.3
- clang_getNullLocation@Base 1:3.3
- clang_getNullRange@Base 1:3.3
- clang_getNumArgTypes@Base 1:3.3
- clang_getNumCompletionChunks@Base 1:3.3
- clang_getNumDiagnostics@Base 1:3.3
- clang_getNumDiagnosticsInSet@Base 1:3.3
- clang_getNumElements@Base 1:3.3
- clang_getNumOverloadedDecls@Base 1:3.3
- clang_getOverloadedDecl@Base 1:3.3
- clang_getOverriddenCursors@Base 1:3.3
- clang_getPointeeType@Base 1:3.3
- clang_getPresumedLocation@Base 1:3.3
- clang_getRange@Base 1:3.3
- clang_getRangeEnd@Base 1:3.3
- clang_getRangeStart@Base 1:3.3
- clang_getRemappings@Base 1:3.3
- clang_getRemappingsFromFileList@Base 1:3.3
- clang_getResultType@Base 1:3.3
- clang_getSpecializedCursorTemplate@Base 1:3.3
- clang_getSpellingLocation@Base 1:3.3
- clang_getTUResourceUsageName@Base 1:3.3
- clang_getTemplateCursorKind@Base 1:3.3
- clang_getTokenExtent@Base 1:3.3
- clang_getTokenKind@Base 1:3.3
- clang_getTokenLocation@Base 1:3.3
- clang_getTokenSpelling@Base 1:3.3
- clang_getTranslationUnitCursor@Base 1:3.3
- clang_getTranslationUnitSpelling@Base 1:3.3
- clang_getTypeDeclaration@Base 1:3.3
- clang_getTypeKindSpelling@Base 1:3.3
- clang_getTypeSpelling@Base 1:3.3
- clang_getTypedefDeclUnderlyingType@Base 1:3.3
- clang_hashCursor@Base 1:3.3
- clang_indexLoc_getCXSourceLocation@Base 1:3.3
- clang_indexLoc_getFileLocation@Base 1:3.3
- clang_indexSourceFile@Base 1:3.3
- clang_indexTranslationUnit@Base 1:3.3
- clang_index_getCXXClassDeclInfo@Base 1:3.3
- clang_index_getClientContainer@Base 1:3.3
- clang_index_getClientEntity@Base 1:3.3
- clang_index_getIBOutletCollectionAttrInfo@Base 1:3.3
- clang_index_getObjCCategoryDeclInfo@Base 1:3.3
- clang_index_getObjCContainerDeclInfo@Base 1:3.3
- clang_index_getObjCInterfaceDeclInfo@Base 1:3.3
- clang_index_getObjCPropertyDeclInfo@Base 1:3.3
- clang_index_getObjCProtocolRefListInfo@Base 1:3.3
- clang_index_isEntityObjCContainerKind@Base 1:3.3
- clang_index_setClientContainer@Base 1:3.3
- clang_index_setClientEntity@Base 1:3.3
- clang_isAttribute@Base 1:3.3
- clang_isConstQualifiedType@Base 1:3.3
- clang_isCursorDefinition@Base 1:3.3
- clang_isDeclaration@Base 1:3.3
- clang_isExpression@Base 1:3.3
- clang_isFileMultipleIncludeGuarded@Base 1:3.3
- clang_isFunctionTypeVariadic@Base 1:3.3
- clang_isInvalid@Base 1:3.3
- clang_isPODType@Base 1:3.3
- clang_isPreprocessing@Base 1:3.3
- clang_isReference@Base 1:3.3
- clang_isRestrictQualifiedType@Base 1:3.3
- clang_isStatement@Base 1:3.3
- clang_isTranslationUnit@Base 1:3.3
- clang_isUnexposed@Base 1:3.3
- clang_isVirtualBase@Base 1:3.3
- clang_isVolatileQualifiedType@Base 1:3.3
- clang_loadDiagnostics@Base 1:3.3
- clang_parseTranslationUnit@Base 1:3.3
- clang_remap_dispose@Base 1:3.3
- clang_remap_getFilenames@Base 1:3.3
- clang_remap_getNumFiles@Base 1:3.3
- clang_reparseTranslationUnit@Base 1:3.3
- clang_saveTranslationUnit@Base 1:3.3
- clang_sortCodeCompletionResults@Base 1:3.3
- clang_toggleCrashRecovery@Base 1:3.3
- clang_tokenize@Base 1:3.3
- clang_visitChildren@Base 1:3.3
- clang_visitChildrenWithBlock@Base 1:3.3
+ clang_BlockCommandComment_getArgText@Base 3.3-debian11ppa2
+ clang_BlockCommandComment_getCommandName@Base 3.3-debian11ppa2
+ clang_BlockCommandComment_getNumArgs@Base 3.3-debian11ppa2
+ clang_BlockCommandComment_getParagraph@Base 3.3-debian11ppa2
+ clang_CXCursorSet_contains@Base 3.3-debian11ppa2
+ clang_CXCursorSet_insert@Base 3.3-debian11ppa2
+ clang_CXIndex_getGlobalOptions@Base 3.3-debian11ppa2
+ clang_CXIndex_setGlobalOptions@Base 3.3-debian11ppa2
+ clang_CXXMethod_isStatic@Base 3.3-debian11ppa2
+ clang_CXXMethod_isVirtual@Base 3.3-debian11ppa2
+ clang_Comment_getChild@Base 3.3-debian11ppa2
+ clang_Comment_getKind@Base 3.3-debian11ppa2
+ clang_Comment_getNumChildren@Base 3.3-debian11ppa2
+ clang_Comment_isWhitespace@Base 3.3-debian11ppa2
+ clang_CompilationDatabase_dispose@Base 3.3-debian11ppa2
+ clang_CompilationDatabase_fromDirectory@Base 3.3-debian11ppa2
+ clang_CompilationDatabase_getAllCompileCommands@Base 3.3-debian11ppa2
+ clang_CompilationDatabase_getCompileCommands@Base 3.3-debian11ppa2
+ clang_CompileCommand_getArg@Base 3.3-debian11ppa2
+ clang_CompileCommand_getDirectory@Base 3.3-debian11ppa2
+ clang_CompileCommand_getNumArgs@Base 3.3-debian11ppa2
+ clang_CompileCommands_dispose@Base 3.3-debian11ppa2
+ clang_CompileCommands_getCommand@Base 3.3-debian11ppa2
+ clang_CompileCommands_getSize@Base 3.3-debian11ppa2
+ clang_Cursor_getArgument@Base 3.3-debian11ppa2
+ clang_Cursor_getBriefCommentText@Base 3.3-debian11ppa2
+ clang_Cursor_getCommentRange@Base 3.3-debian11ppa2
+ clang_Cursor_getModule@Base 3.3-debian11ppa2
+ clang_Cursor_getNumArguments@Base 3.3-debian11ppa2
+ clang_Cursor_getObjCDeclQualifiers@Base 3.3-debian11ppa2
+ clang_Cursor_getObjCPropertyAttributes@Base 3.3-debian11ppa2
+ clang_Cursor_getObjCSelectorIndex@Base 3.3-debian11ppa2
+ clang_Cursor_getParsedComment@Base 3.3-debian11ppa2
+ clang_Cursor_getRawCommentText@Base 3.3-debian11ppa2
+ clang_Cursor_getReceiverType@Base 3.3-debian11ppa2
+ clang_Cursor_getSpellingNameRange@Base 3.3-debian11ppa2
+ clang_Cursor_getTranslationUnit@Base 3.3-debian11ppa2
+ clang_Cursor_isBitField@Base 3.3-debian11ppa2
+ clang_Cursor_isDynamicCall@Base 3.3-debian11ppa2
+ clang_Cursor_isNull@Base 3.3-debian11ppa2
+ clang_Cursor_isVariadic@Base 3.3-debian11ppa2
+ clang_FullComment_getAsHTML@Base 3.3-debian11ppa2
+ clang_FullComment_getAsXML@Base 3.3-debian11ppa2
+ clang_HTMLStartTagComment_isSelfClosing@Base 3.3-debian11ppa2
+ clang_HTMLStartTag_getAttrName@Base 3.3-debian11ppa2
+ clang_HTMLStartTag_getAttrValue@Base 3.3-debian11ppa2
+ clang_HTMLStartTag_getNumAttrs@Base 3.3-debian11ppa2
+ clang_HTMLTagComment_getAsString@Base 3.3-debian11ppa2
+ clang_HTMLTagComment_getTagName@Base 3.3-debian11ppa2
+ clang_IndexAction_create@Base 3.3-debian11ppa2
+ clang_IndexAction_dispose@Base 3.3-debian11ppa2
+ clang_InlineCommandComment_getArgText@Base 3.3-debian11ppa2
+ clang_InlineCommandComment_getCommandName@Base 3.3-debian11ppa2
+ clang_InlineCommandComment_getNumArgs@Base 3.3-debian11ppa2
+ clang_InlineCommandComment_getRenderKind@Base 3.3-debian11ppa2
+ clang_InlineContentComment_hasTrailingNewline@Base 3.3-debian11ppa2
+ clang_Location_isInSystemHeader@Base 3.3-debian11ppa2
+ clang_Module_getASTFile@Base 3.3-debian11ppa2
+ clang_Module_getFullName@Base 3.3-debian11ppa2
+ clang_Module_getName@Base 3.3-debian11ppa2
+ clang_Module_getNumTopLevelHeaders@Base 3.3-debian11ppa2
+ clang_Module_getParent@Base 3.3-debian11ppa2
+ clang_Module_getTopLevelHeader@Base 3.3-debian11ppa2
+ clang_ParamCommandComment_getDirection@Base 3.3-debian11ppa2
+ clang_ParamCommandComment_getParamIndex@Base 3.3-debian11ppa2
+ clang_ParamCommandComment_getParamName@Base 3.3-debian11ppa2
+ clang_ParamCommandComment_isDirectionExplicit@Base 3.3-debian11ppa2
+ clang_ParamCommandComment_isParamIndexValid@Base 3.3-debian11ppa2
+ clang_Range_isNull@Base 3.3-debian11ppa2
+ clang_TParamCommandComment_getDepth@Base 3.3-debian11ppa2
+ clang_TParamCommandComment_getIndex@Base 3.3-debian11ppa2
+ clang_TParamCommandComment_getParamName@Base 3.3-debian11ppa2
+ clang_TParamCommandComment_isParamPositionValid@Base 3.3-debian11ppa2
+ clang_TextComment_getText@Base 3.3-debian11ppa2
+ clang_Type_getAlignOf@Base 3.3-debian11ppa2
+ clang_Type_getOffsetOf@Base 3.3-debian11ppa2
+ clang_Type_getSizeOf@Base 3.3-debian11ppa2
+ clang_VerbatimBlockLineComment_getText@Base 3.3-debian11ppa2
+ clang_VerbatimLineComment_getText@Base 3.3-debian11ppa2
+ clang_annotateTokens@Base 3.3-debian11ppa2
+ clang_codeCompleteAt@Base 3.3-debian11ppa2
+ clang_codeCompleteGetContainerKind@Base 3.3-debian11ppa2
+ clang_codeCompleteGetContainerUSR@Base 3.3-debian11ppa2
+ clang_codeCompleteGetContexts@Base 3.3-debian11ppa2
+ clang_codeCompleteGetDiagnostic@Base 3.3-debian11ppa2
+ clang_codeCompleteGetNumDiagnostics@Base 3.3-debian11ppa2
+ clang_codeCompleteGetObjCSelector@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCCategory@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCClass@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCIvar@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCMethod@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCProperty@Base 3.3-debian11ppa2
+ clang_constructUSR_ObjCProtocol@Base 3.3-debian11ppa2
+ clang_createCXCursorSet@Base 3.3-debian11ppa2
+ clang_createIndex@Base 3.3-debian11ppa2
+ clang_createTranslationUnit@Base 3.3-debian11ppa2
+ clang_createTranslationUnitFromSourceFile@Base 3.3-debian11ppa2
+ clang_defaultCodeCompleteOptions@Base 3.3-debian11ppa2
+ clang_defaultDiagnosticDisplayOptions@Base 3.3-debian11ppa2
+ clang_defaultEditingTranslationUnitOptions@Base 3.3-debian11ppa2
+ clang_defaultReparseOptions@Base 3.3-debian11ppa2
+ clang_defaultSaveOptions@Base 3.3-debian11ppa2
+ clang_disposeCXCursorSet@Base 3.3-debian11ppa2
+ clang_disposeCXPlatformAvailability@Base 3.3-debian11ppa2
+ clang_disposeCXTUResourceUsage@Base 3.3-debian11ppa2
+ clang_disposeCodeCompleteResults@Base 3.3-debian11ppa2
+ clang_disposeDiagnostic@Base 3.3-debian11ppa2
+ clang_disposeDiagnosticSet@Base 3.3-debian11ppa2
+ clang_disposeIndex@Base 3.3-debian11ppa2
+ clang_disposeOverriddenCursors@Base 3.3-debian11ppa2
+ clang_disposeString@Base 3.3-debian11ppa2
+ clang_disposeTokens@Base 3.3-debian11ppa2
+ clang_disposeTranslationUnit@Base 3.3-debian11ppa2
+ clang_enableStackTraces@Base 3.3-debian11ppa2
+ clang_equalCursors@Base 3.3-debian11ppa2
+ clang_equalLocations@Base 3.3-debian11ppa2
+ clang_equalRanges@Base 3.3-debian11ppa2
+ clang_equalTypes@Base 3.3-debian11ppa2
+ clang_executeOnThread@Base 3.3-debian11ppa2
+ clang_findIncludesInFile@Base 3.3-debian11ppa2
+ clang_findIncludesInFileWithBlock@Base 3.3-debian11ppa2
+ clang_findReferencesInFile@Base 3.3-debian11ppa2
+ clang_findReferencesInFileWithBlock@Base 3.3-debian11ppa2
+ clang_formatDiagnostic@Base 3.3-debian11ppa2
+ clang_getArgType@Base 3.3-debian11ppa2
+ clang_getArrayElementType@Base 3.3-debian11ppa2
+ clang_getArraySize@Base 3.3-debian11ppa2
+ clang_getCString@Base 3.3-debian11ppa2
+ clang_getCXTUResourceUsage@Base 3.3-debian11ppa2
+ clang_getCXXAccessSpecifier@Base 3.3-debian11ppa2
+ clang_getCanonicalCursor@Base 3.3-debian11ppa2
+ clang_getCanonicalType@Base 3.3-debian11ppa2
+ clang_getChildDiagnostics@Base 3.3-debian11ppa2
+ clang_getClangVersion@Base 3.3-debian11ppa2
+ clang_getCompletionAnnotation@Base 3.3-debian11ppa2
+ clang_getCompletionAvailability@Base 3.3-debian11ppa2
+ clang_getCompletionBriefComment@Base 3.3-debian11ppa2
+ clang_getCompletionChunkCompletionString@Base 3.3-debian11ppa2
+ clang_getCompletionChunkKind@Base 3.3-debian11ppa2
+ clang_getCompletionChunkText@Base 3.3-debian11ppa2
+ clang_getCompletionNumAnnotations@Base 3.3-debian11ppa2
+ clang_getCompletionParent@Base 3.3-debian11ppa2
+ clang_getCompletionPriority@Base 3.3-debian11ppa2
+ clang_getCursor@Base 3.3-debian11ppa2
+ clang_getCursorAvailability@Base 3.3-debian11ppa2
+ clang_getCursorCompletionString@Base 3.3-debian11ppa2
+ clang_getCursorDefinition@Base 3.3-debian11ppa2
+ clang_getCursorDisplayName@Base 3.3-debian11ppa2
+ clang_getCursorExtent@Base 3.3-debian11ppa2
+ clang_getCursorKind@Base 3.3-debian11ppa2
+ clang_getCursorKindSpelling@Base 3.3-debian11ppa2
+ clang_getCursorLanguage@Base 3.3-debian11ppa2
+ clang_getCursorLexicalParent@Base 3.3-debian11ppa2
+ clang_getCursorLinkage@Base 3.3-debian11ppa2
+ clang_getCursorLocation@Base 3.3-debian11ppa2
+ clang_getCursorPlatformAvailability@Base 3.3-debian11ppa2
+ clang_getCursorReferenceNameRange@Base 3.3-debian11ppa2
+ clang_getCursorReferenced@Base 3.3-debian11ppa2
+ clang_getCursorResultType@Base 3.3-debian11ppa2
+ clang_getCursorSemanticParent@Base 3.3-debian11ppa2
+ clang_getCursorSpelling@Base 3.3-debian11ppa2
+ clang_getCursorType@Base 3.3-debian11ppa2
+ clang_getCursorUSR@Base 3.3-debian11ppa2
+ clang_getDeclObjCTypeEncoding@Base 3.3-debian11ppa2
+ clang_getDefinitionSpellingAndExtent@Base 3.3-debian11ppa2
+ clang_getDiagnostic@Base 3.3-debian11ppa2
+ clang_getDiagnosticCategory@Base 3.3-debian11ppa2
+ clang_getDiagnosticCategoryName@Base 3.3-debian11ppa2
+ clang_getDiagnosticCategoryText@Base 3.3-debian11ppa2
+ clang_getDiagnosticFixIt@Base 3.3-debian11ppa2
+ clang_getDiagnosticInSet@Base 3.3-debian11ppa2
+ clang_getDiagnosticLocation@Base 3.3-debian11ppa2
+ clang_getDiagnosticNumFixIts@Base 3.3-debian11ppa2
+ clang_getDiagnosticNumRanges@Base 3.3-debian11ppa2
+ clang_getDiagnosticOption@Base 3.3-debian11ppa2
+ clang_getDiagnosticRange@Base 3.3-debian11ppa2
+ clang_getDiagnosticSetFromTU@Base 3.3-debian11ppa2
+ clang_getDiagnosticSeverity@Base 3.3-debian11ppa2
+ clang_getDiagnosticSpelling@Base 3.3-debian11ppa2
+ clang_getElementType@Base 3.3-debian11ppa2
+ clang_getEnumConstantDeclUnsignedValue@Base 3.3-debian11ppa2
+ clang_getEnumConstantDeclValue@Base 3.3-debian11ppa2
+ clang_getEnumDeclIntegerType@Base 3.3-debian11ppa2
+ clang_getExpansionLocation@Base 3.3-debian11ppa2
+ clang_getFieldDeclBitWidth@Base 3.3-debian11ppa2
+ clang_getFile@Base 3.3-debian11ppa2
+ clang_getFileLocation@Base 3.3-debian11ppa2
+ clang_getFileName@Base 3.3-debian11ppa2
+ clang_getFileTime@Base 3.3-debian11ppa2
+ clang_getFileUniqueID@Base 3.3-debian11ppa2
+ clang_getFunctionTypeCallingConv@Base 3.3-debian11ppa2
+ clang_getIBOutletCollectionType@Base 3.3-debian11ppa2
+ clang_getIncludedFile@Base 3.3-debian11ppa2
+ clang_getInclusions@Base 3.3-debian11ppa2
+ clang_getInstantiationLocation@Base 3.3-debian11ppa2
+ clang_getLocation@Base 3.3-debian11ppa2
+ clang_getLocationForOffset@Base 3.3-debian11ppa2
+ clang_getNullCursor@Base 3.3-debian11ppa2
+ clang_getNullLocation@Base 3.3-debian11ppa2
+ clang_getNullRange@Base 3.3-debian11ppa2
+ clang_getNumArgTypes@Base 3.3-debian11ppa2
+ clang_getNumCompletionChunks@Base 3.3-debian11ppa2
+ clang_getNumDiagnostics@Base 3.3-debian11ppa2
+ clang_getNumDiagnosticsInSet@Base 3.3-debian11ppa2
+ clang_getNumElements@Base 3.3-debian11ppa2
+ clang_getNumOverloadedDecls@Base 3.3-debian11ppa2
+ clang_getOverloadedDecl@Base 3.3-debian11ppa2
+ clang_getOverriddenCursors@Base 3.3-debian11ppa2
+ clang_getPointeeType@Base 3.3-debian11ppa2
+ clang_getPresumedLocation@Base 3.3-debian11ppa2
+ clang_getRange@Base 3.3-debian11ppa2
+ clang_getRangeEnd@Base 3.3-debian11ppa2
+ clang_getRangeStart@Base 3.3-debian11ppa2
+ clang_getRemappings@Base 3.3-debian11ppa2
+ clang_getRemappingsFromFileList@Base 3.3-debian11ppa2
+ clang_getResultType@Base 3.3-debian11ppa2
+ clang_getSpecializedCursorTemplate@Base 3.3-debian11ppa2
+ clang_getSpellingLocation@Base 3.3-debian11ppa2
+ clang_getTUResourceUsageName@Base 3.3-debian11ppa2
+ clang_getTemplateCursorKind@Base 3.3-debian11ppa2
+ clang_getTokenExtent@Base 3.3-debian11ppa2
+ clang_getTokenKind@Base 3.3-debian11ppa2
+ clang_getTokenLocation@Base 3.3-debian11ppa2
+ clang_getTokenSpelling@Base 3.3-debian11ppa2
+ clang_getTranslationUnitCursor@Base 3.3-debian11ppa2
+ clang_getTranslationUnitSpelling@Base 3.3-debian11ppa2
+ clang_getTypeDeclaration@Base 3.3-debian11ppa2
+ clang_getTypeKindSpelling@Base 3.3-debian11ppa2
+ clang_getTypeSpelling@Base 3.3-debian11ppa2
+ clang_getTypedefDeclUnderlyingType@Base 3.3-debian11ppa2
+ clang_hashCursor@Base 3.3-debian11ppa2
+ clang_indexLoc_getCXSourceLocation@Base 3.3-debian11ppa2
+ clang_indexLoc_getFileLocation@Base 3.3-debian11ppa2
+ clang_indexSourceFile@Base 3.3-debian11ppa2
+ clang_indexTranslationUnit@Base 3.3-debian11ppa2
+ clang_index_getCXXClassDeclInfo@Base 3.3-debian11ppa2
+ clang_index_getClientContainer@Base 3.3-debian11ppa2
+ clang_index_getClientEntity@Base 3.3-debian11ppa2
+ clang_index_getIBOutletCollectionAttrInfo@Base 3.3-debian11ppa2
+ clang_index_getObjCCategoryDeclInfo@Base 3.3-debian11ppa2
+ clang_index_getObjCContainerDeclInfo@Base 3.3-debian11ppa2
+ clang_index_getObjCInterfaceDeclInfo@Base 3.3-debian11ppa2
+ clang_index_getObjCPropertyDeclInfo@Base 3.3-debian11ppa2
+ clang_index_getObjCProtocolRefListInfo@Base 3.3-debian11ppa2
+ clang_index_isEntityObjCContainerKind@Base 3.3-debian11ppa2
+ clang_index_setClientContainer@Base 3.3-debian11ppa2
+ clang_index_setClientEntity@Base 3.3-debian11ppa2
+ clang_isAttribute@Base 3.3-debian11ppa2
+ clang_isConstQualifiedType@Base 3.3-debian11ppa2
+ clang_isCursorDefinition@Base 3.3-debian11ppa2
+ clang_isDeclaration@Base 3.3-debian11ppa2
+ clang_isExpression@Base 3.3-debian11ppa2
+ clang_isFileMultipleIncludeGuarded@Base 3.3-debian11ppa2
+ clang_isFunctionTypeVariadic@Base 3.3-debian11ppa2
+ clang_isInvalid@Base 3.3-debian11ppa2
+ clang_isPODType@Base 3.3-debian11ppa2
+ clang_isPreprocessing@Base 3.3-debian11ppa2
+ clang_isReference@Base 3.3-debian11ppa2
+ clang_isRestrictQualifiedType@Base 3.3-debian11ppa2
+ clang_isStatement@Base 3.3-debian11ppa2
+ clang_isTranslationUnit@Base 3.3-debian11ppa2
+ clang_isUnexposed@Base 3.3-debian11ppa2
+ clang_isVirtualBase@Base 3.3-debian11ppa2
+ clang_isVolatileQualifiedType@Base 3.3-debian11ppa2
+ clang_loadDiagnostics@Base 3.3-debian11ppa2
+ clang_parseTranslationUnit@Base 3.3-debian11ppa2
+ clang_remap_dispose@Base 3.3-debian11ppa2
+ clang_remap_getFilenames@Base 3.3-debian11ppa2
+ clang_remap_getNumFiles@Base 3.3-debian11ppa2
+ clang_reparseTranslationUnit@Base 3.3-debian11ppa2
+ clang_saveTranslationUnit@Base 3.3-debian11ppa2
+ clang_sortCodeCompletionResults@Base 3.3-debian11ppa2
+ clang_toggleCrashRecovery@Base 3.3-debian11ppa2
+ clang_tokenize@Base 3.3-debian11ppa2
+ clang_visitChildren@Base 3.3-debian11ppa2
+ clang_visitChildrenWithBlock@Base 3.3-debian11ppa2
diff -urN debian/libllvm-3.3-ocaml-dev.dirs.in llvm-toolchain-snapshot_3.3~svn183554/debian/libllvm-3.3-ocaml-dev.dirs.in
--- debian/libllvm-3.3-ocaml-dev.dirs.in	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/libllvm-3.3-ocaml-dev.dirs.in	2013-09-04 18:21:38.839126659 +0900
@@ -1,2 +1,2 @@
-@OCAML_STDLIB_DIR@/META/
+@OCAML_STDLIB_DIR@/METAS
 
diff -urN debian/llvm-3.3-dev.install llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-dev.install
--- debian/llvm-3.3-dev.install	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-dev.install	2013-09-04 18:21:38.839126659 +0900
@@ -13,8 +13,8 @@
 configure			            /usr/lib/llvm-3.3/build/
 autoconf/			            /usr/lib/llvm-3.3/build/
 
-utils/vim/llvm-3.3.vim		        usr/share/vim/addons/syntax/
-utils/vim/tablegen-3.3.vim		    usr/share/vim/addons/syntax/
+utils/vim/llvm.vim                      usr/share/vim/addons/syntax/llvm-3.3.vim
+utils/vim/tablegen.vim                      usr/share/vim/addons/syntax/tablegen-3.3.vim
 
 utils/emacs/emacs.el		    usr/share/emacs/site-lisp/llvm-3.3/
 utils/emacs/llvm-mode.el	    usr/share/emacs/site-lisp/llvm-3.3/
diff -urN debian/llvm-3.3-runtime.postinst llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-runtime.postinst
--- debian/llvm-3.3-runtime.postinst	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-runtime.postinst	1970-01-01 09:00:00.000000000 +0900
@@ -1,9 +0,0 @@
-#!/bin/sh -e
-
-if test "$1" = "configure"; then
-    if test -x /usr/sbin/update-binfmts; then
-        update-binfmts --import llvm-3.3-runtime.binfmt || true
-    fi
-fi
-
-#DEBHELPER#
diff -urN debian/llvm-3.3-runtime.prerm.in llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-runtime.prerm.in
--- debian/llvm-3.3-runtime.prerm.in	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/llvm-3.3-runtime.prerm.in	1970-01-01 09:00:00.000000000 +0900
@@ -1,16 +0,0 @@
-#!/bin/sh -e
-
-if test "$1" = "remove"; then
-    if test -x /usr/sbin/update-binfmts; then
-        update-binfmts --package llvm-@LLVM_VERSION@-runtime \
-            --remove llvm-@LLVM_VERSION@-runtime.binfmt /usr/bin/lli-@LLVM_VERSION@ || true
-        if test -f /var/lib/binfmts/llvm-@LLVM_VERSION@.binfmt; then
-            # Purge old file
-            update-binfmts --package llvm-@LLVM_VERSION@-runtime \
-                --remove llvm-@LLVM_VERSION@.binfmt /usr/bin/lli-@LLVM_VERSION@ || true
-        fi
-    fi
-fi
-
-#DEBHELPER#
-
diff -urN debian/man/clang-format-diff-3.3.1 llvm-toolchain-snapshot_3.3~svn183554/debian/man/clang-format-diff-3.3.1
--- debian/man/clang-format-diff-3.3.1	2013-08-01 21:04:41.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/man/clang-format-diff-3.3.1	2013-12-19 21:50:10.253154939 +0900
@@ -1,9 +1,9 @@
-.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.41.2.
-.TH CLANG-FORMAT-DIFF-3.3 "1" "June 2013" "clang-format-diff-3.3 3.3" "User Commands"
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.4.
+.TH CLANG-FORMAT-DIFF.PY "1" "December 2013" "clang-format-diff.py 3.3" "User Commands"
 .SH NAME
-clang-format-diff-3.3 \- manual page for clang-format-diff-3.3 3.3
+clang-format-diff.py \- manual page for clang-format-diff.py 3.3
 .SH DESCRIPTION
-usage: clang\-format\-diff\-3.3 [\-h] [\-p P] [\-style STYLE]
+usage: clang\-format\-diff.py [\-h] [\-p P] [\-style STYLE]
 .PP
 Reformat changed lines in diff
 .SS "optional arguments:"
@@ -18,13 +18,13 @@
 formatting style to apply (LLVM, Google, Chromium)
 .SH "SEE ALSO"
 The full documentation for
-.B clang-format-diff-3.3
+.B clang-format-diff.py
 is maintained as a Texinfo manual.  If the
 .B info
 and
-.B clang-format-diff-3.3
+.B clang-format-diff.py
 programs are properly installed at your site, the command
 .IP
-.B info clang-format-diff-3.3
+.B info clang-format-diff.py
 .PP
 should give you access to the complete manual.
diff -urN debian/patches/28-gcc-4.7-paths.diff llvm-toolchain-snapshot_3.3~svn183554/debian/patches/28-gcc-4.7-paths.diff
--- debian/patches/28-gcc-4.7-paths.diff	1970-01-01 09:00:00.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/patches/28-gcc-4.7-paths.diff	2013-09-04 18:21:38.839126659 +0900
@@ -0,0 +1,48 @@
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/lib/Driver/ToolChains.cpp llvm-toolchain-snapshot_3.3~svn183554/clang/lib/Driver/ToolChains.cpp
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/lib/Driver/ToolChains.cpp	2013-07-03 00:06:02.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/lib/Driver/ToolChains.cpp	2013-07-08 13:56:39.057345859 +0900
+@@ -36,6 +36,10 @@
+ 
+ #include <cstdlib> // ::getenv
+ 
++#include "clang/Debian/debian_path.h"
++
++#include <iostream>
++
+ using namespace clang::driver;
+ using namespace clang::driver::toolchains;
+ using namespace clang;
+@@ -2574,14 +2578,12 @@
+   StringRef Version = GCCInstallation.getVersion().Text;
+   StringRef TripleStr = GCCInstallation.getTriple().str();
+ 
+-  if (addLibStdCXXIncludePaths(LibDir.str() + "/../include", 
+-                               "/c++/" + Version.str(),
+-                               TripleStr,
+-                               GCCInstallation.getMultiarchSuffix(),
+-                               DriverArgs, CC1Args))
+-    return;
+-
+   const std::string IncludePathCandidates[] = {
++    LibDir.str() + "/../include/c++/" + Version.str(),
++    // Once more, Debian headers locations changed (since gcc version 4.7.2-5)
++    // See Debian bug tracker #693240
++    LibDir.str() + "/../include/"+ TripleStr.str() +"/c++/" + Version.str(),
++    LibDir.str() + "/../include/" + DEB_HOST_MULTIARCH_TRIPLET + "/c++/" + Version.str(),
+     // Gentoo is weird and places its headers inside the GCC install, so if the
+     // first attempt to find the headers fails, try this pattern.
+     InstallDir.str() + "/include/g++-v4",
+@@ -2593,10 +2595,10 @@
+   };
+ 
+   for (unsigned i = 0; i < llvm::array_lengthof(IncludePathCandidates); ++i) {
+-    if (addLibStdCXXIncludePaths(IncludePathCandidates[i], (TripleStr +
++      addLibStdCXXIncludePaths(IncludePathCandidates[i], (TripleStr +
+                 GCCInstallation.getMultiarchSuffix()),
+-            DriverArgs, CC1Args))
+-      break;
++                               DriverArgs, CC1Args);
++
+   }
+ }
+ 
diff -urN debian/patches/98-unwind.h.patch llvm-toolchain-snapshot_3.3~svn183554/debian/patches/98-unwind.h.patch
--- debian/patches/98-unwind.h.patch	1970-01-01 09:00:00.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/patches/98-unwind.h.patch	2013-09-04 18:21:38.839126659 +0900
@@ -0,0 +1,22 @@
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/lib/Headers/unwind.h	2013-01-17 08:08:36.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/lib/Headers/unwind.h	2013-07-02 01:55:23.756641885 +0900
+@@ -24,7 +24,6 @@
+ /* See "Data Definitions for libgcc_s" in the Linux Standard Base.*/
+ 
+ #ifndef __CLANG_UNWIND_H
+-#define __CLANG_UNWIND_H
+ 
+ #if __has_include_next(<unwind.h>)
+ /* Darwin and libunwind provide an unwind.h. If that's available, use
+@@ -51,8 +50,11 @@
+ #  undef _GNU_SOURCE
+ #  undef _SHOULD_UNDEFINE_GNU_SOURCE
+ # endif
++#define __CLANG_UNWIND_H
+ #else
+ 
++#define __CLANG_UNWIND_H
++
+ #include <stdint.h>
+ 
+ #ifdef __cplusplus
diff -urN debian/patches/99-python-bindings.patch llvm-toolchain-snapshot_3.3~svn183554/debian/patches/99-python-bindings.patch
--- debian/patches/99-python-bindings.patch	1970-01-01 09:00:00.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/patches/99-python-bindings.patch	2013-10-18 23:25:00.455118831 +0900
@@ -0,0 +1,235 @@
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/README.txt llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/README.txt
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/README.txt	2010-05-01 06:51:10.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/README.txt	2013-10-18 23:21:31.714083758 +0900
+@@ -4,12 +4,9 @@
+ 
+ This directory implements Python bindings for Clang.
+ 
+-You may need to alter LD_LIBRARY_PATH so that the Clang library can be
+-found. The unit tests are designed to be run with 'nosetests'. For example:
++The unit tests are designed to be run with 'nosetests'. For example:
+ --
+-$ env PYTHONPATH=$(echo ~/llvm/tools/clang/bindings/python/) \
+-      LD_LIBRARY_PATH=$(llvm-config --libdir) \
+-  nosetests -v
++$ nosetests -v
+ tests.cindex.test_index.test_create ... ok
+ ...
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/clang/cindex.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/clang/cindex.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/clang/cindex.py	2013-04-22 03:35:51.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/clang/cindex.py	2013-10-18 23:23:49.494766979 +0900
+@@ -65,7 +65,7 @@
+ from ctypes import *
+ import collections
+ 
+-import clang.enumerations
++import clang33.enumerations
+ 
+ # ctypes doesn't implicitly convert c_void_p to the appropriate wrapper
+ # object. This is a problem, because it means that from_parameter will see an
+@@ -3224,7 +3224,7 @@
+         return True
+ 
+ def register_enumerations():
+-    for name, value in clang.enumerations.TokenKinds:
++    for name, value in clang33.enumerations.TokenKinds:
+         TokenKind.register(value, name)
+ 
+ conf = Config()
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/examples/cindex/cindex-dump.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/examples/cindex/cindex-dump.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/examples/cindex/cindex-dump.py	2010-02-14 03:33:28.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/examples/cindex/cindex-dump.py	2013-10-18 23:21:31.714083758 +0900
+@@ -54,7 +54,7 @@
+              'children' : children }
+ 
+ def main():
+-    from clang.cindex import Index
++    from clang33.cindex import Index
+     from pprint import pprint
+ 
+     from optparse import OptionParser, OptionGroup
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/examples/cindex/cindex-includes.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/examples/cindex/cindex-includes.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/examples/cindex/cindex-includes.py	2010-02-14 03:33:28.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/examples/cindex/cindex-includes.py	2013-10-18 23:21:31.718083762 +0900
+@@ -16,7 +16,7 @@
+ 
+ def main():
+     import sys
+-    from clang.cindex import Index
++    from clang33.cindex import Index
+ 
+     from optparse import OptionParser, OptionGroup
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cdb.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cdb.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cdb.py	2012-07-10 09:00:05.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cdb.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,7 +1,7 @@
+-from clang.cindex import CompilationDatabase
+-from clang.cindex import CompilationDatabaseError
+-from clang.cindex import CompileCommands
+-from clang.cindex import CompileCommand
++from clang33.cindex import CompilationDatabase
++from clang33.cindex import CompilationDatabaseError
++from clang33.cindex import CompileCommands
++from clang33.cindex import CompileCommand
+ import os
+ import gc
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_code_completion.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_code_completion.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_code_completion.py	2012-10-07 20:46:37.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_code_completion.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import TranslationUnit
++from clang33.cindex import TranslationUnit
+ 
+ def check_completion_results(cr, expected):
+     assert cr is not None
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cursor.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cursor.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cursor.py	2013-01-03 07:31:57.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cursor.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,8 +1,8 @@
+ import gc
+ 
+-from clang.cindex import CursorKind
+-from clang.cindex import TranslationUnit
+-from clang.cindex import TypeKind
++from clang33.cindex import CursorKind
++from clang33.cindex import TranslationUnit
++from clang33.cindex import TypeKind
+ from .util import get_cursor
+ from .util import get_cursors
+ from .util import get_tu
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cursor_kind.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cursor_kind.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_cursor_kind.py	2012-02-05 20:42:09.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_cursor_kind.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import CursorKind
++from clang33.cindex import CursorKind
+ 
+ def test_name():
+     assert CursorKind.UNEXPOSED_DECL.name is 'UNEXPOSED_DECL'
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_diagnostics.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_diagnostics.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_diagnostics.py	2012-03-11 07:19:05.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_diagnostics.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang33.cindex import *
+ from .util import get_tu
+ 
+ # FIXME: We need support for invalid translation units to test better.
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_file.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_file.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_file.py	2011-10-31 09:07:19.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_file.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import Index, File
++from clang33.cindex import Index, File
+ 
+ def test_file():
+   index = Index.create()
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_index.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_index.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_index.py	2010-01-24 11:02:07.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_index.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang33.cindex import *
+ import os
+ 
+ kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_location.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_location.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_location.py	2012-06-11 20:11:48.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_location.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,7 +1,7 @@
+-from clang.cindex import Cursor
+-from clang.cindex import File
+-from clang.cindex import SourceLocation
+-from clang.cindex import SourceRange
++from clang33.cindex import Cursor
++from clang33.cindex import File
++from clang33.cindex import SourceLocation
++from clang33.cindex import SourceRange
+ from .util import get_cursor
+ from .util import get_tu
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_token_kind.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_token_kind.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_token_kind.py	2012-07-12 16:21:12.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_token_kind.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import TokenKind
++from clang33.cindex import TokenKind
+ from nose.tools import eq_
+ from nose.tools import ok_
+ from nose.tools import raises
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_tokens.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_tokens.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_tokens.py	2012-07-12 16:21:12.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_tokens.py	2013-10-18 23:21:31.718083762 +0900
+@@ -1,8 +1,8 @@
+-from clang.cindex import CursorKind
+-from clang.cindex import Index
+-from clang.cindex import SourceLocation
+-from clang.cindex import SourceRange
+-from clang.cindex import TokenKind
++from clang33.cindex import CursorKind
++from clang33.cindex import Index
++from clang33.cindex import SourceLocation
++from clang33.cindex import SourceRange
++from clang33.cindex import TokenKind
+ from nose.tools import eq_
+ from nose.tools import ok_
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_translation_unit.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_translation_unit.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-03-20 00:30:48.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-10-18 23:21:31.722083767 +0900
+@@ -1,15 +1,15 @@
+ import gc
+ import os
+ 
+-from clang.cindex import CursorKind
+-from clang.cindex import Cursor
+-from clang.cindex import File
+-from clang.cindex import Index
+-from clang.cindex import SourceLocation
+-from clang.cindex import SourceRange
+-from clang.cindex import TranslationUnitSaveError
+-from clang.cindex import TranslationUnitLoadError
+-from clang.cindex import TranslationUnit
++from clang33.cindex import CursorKind
++from clang33.cindex import Cursor
++from clang33.cindex import File
++from clang33.cindex import Index
++from clang33.cindex import SourceLocation
++from clang33.cindex import SourceRange
++from clang33.cindex import TranslationUnitSaveError
++from clang33.cindex import TranslationUnitLoadError
++from clang33.cindex import TranslationUnit
+ from .util import get_cursor
+ from .util import get_tu
+ 
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_type.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_type.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/test_type.py	2013-04-11 10:20:11.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/test_type.py	2013-10-18 23:21:31.722083767 +0900
+@@ -1,8 +1,8 @@
+ import gc
+ 
+-from clang.cindex import CursorKind
+-from clang.cindex import TranslationUnit
+-from clang.cindex import TypeKind
++from clang33.cindex import CursorKind
++from clang33.cindex import TranslationUnit
++from clang33.cindex import TypeKind
+ from nose.tools import raises
+ from .util import get_cursor
+ from .util import get_tu
+diff -urN llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/util.py llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/util.py
+--- llvm-toolchain-snapshot_3.3~svn183554.orig/clang/bindings/python/tests/cindex/util.py	2013-04-11 10:20:11.000000000 +0900
++++ llvm-toolchain-snapshot_3.3~svn183554/clang/bindings/python/tests/cindex/util.py	2013-10-18 23:21:31.722083767 +0900
+@@ -1,7 +1,7 @@
+ # This file provides common utility functions for the test suite.
+ 
+-from clang.cindex import Cursor
+-from clang.cindex import TranslationUnit
++from clang33.cindex import Cursor
++from clang33.cindex import TranslationUnit
+ 
+ def get_tu(source, lang='c', all_warnings=False, flags=[]):
+     """Obtain a translation unit from source and language.
diff -urN debian/patches/series llvm-toolchain-snapshot_3.3~svn183554/debian/patches/series
--- debian/patches/series	2013-12-01 00:37:53.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/patches/series	2013-12-11 23:12:42.404538389 +0900
@@ -6,6 +6,7 @@
 27-fix_clang_stdint.diff
 26-set-correct-float-abi.diff
 profile_rt.diff
+#28-gcc-4.7-paths.diff
 30-kfreebsd.diff
 0003-Debian-version-info-and-bugreport.patch
 0021-shared-lib-debian.patch
@@ -41,3 +42,5 @@
 removeduplicatedeclaration.diff
 python-clangpath.diff
 debian-mips64-n32-multiarch.diff
+#98-unwind.h.patch
+99-python-bindings.patch
diff -urN debian/pollycc.in llvm-toolchain-snapshot_3.3~svn183554/debian/pollycc.in
--- debian/pollycc.in	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/pollycc.in	2013-09-04 18:21:38.843126677 +0900
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-clang -Xclang -load -Xclang /usr/lib/llvm-@LLVM_VERSION@/lib/LLVMPolly.so $@
+clang-@LLVM_VERSION@ -Xclang -load -Xclang /usr/lib/llvm-@LLVM_VERSION@/lib/LLVMPolly.so $@
diff -urN debian/python-clang-3.3.install llvm-toolchain-snapshot_3.3~svn183554/debian/python-clang-3.3.install
--- debian/python-clang-3.3.install	2013-10-14 19:49:12.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/python-clang-3.3.install	2013-10-18 20:34:20.772736054 +0900
@@ -1 +1,4 @@
-tools/clang/bindings/python/clang/ /usr/lib/python2.7/dist-packages/
+clang/bindings/python/clang/*		/usr/lib/python2.7/dist-packages/clang33/
+clang/bindings/python/README.txt	/usr/share/doc/python-clang-3.3/
+clang/bindings/python/examples		/usr/share/doc/python-clang-3.3/
+clang/bindings/python/tests		/usr/share/doc/python-clang-3.3/
diff -urN debian/rules llvm-toolchain-snapshot_3.3~svn183554/debian/rules
--- debian/rules	2013-12-05 01:51:37.000000000 +0900
+++ llvm-toolchain-snapshot_3.3~svn183554/debian/rules	2013-12-12 02:05:19.253949249 +0900
@@ -1,7 +1,7 @@
 #!/usr/bin/make -f
 
 TARGET_BUILD	:= build-llvm
-GCC_VERSION     := $(shell dpkg-query -W -f '$${Version}' g++ | sed -rne 's,^([0-9]+:)?([0-9]+\.[0-9]+).*$$,\2,p')
+GCC_VERSION     := 4.6
 LLVM_VERSION	:= 3.3
 SONAME_EXT      := 1
 DEBIAN_REVISION := $(shell dpkg-parsechangelog | sed -rne 's,^Version: .*-(.*),\1,p')
@@ -9,7 +9,7 @@
   NJOBS := -j $(subst parallel=,,$(filter parallel=%,$(subst $(COMMA), ,$(DEB_BUILD_OPTIONS))))
 endif
 
-VENDOR=$(shell lsb_release -is)
+VENDOR=Debian
 
 DEB_HOST_MULTIARCH  = $(shell dpkg-architecture -qDEB_HOST_MULTIARCH)
 DEB_HOST_GNU_TYPE   = $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
@@ -79,20 +79,14 @@
         LLDB_ENABLE=no
 endif
 
-
 %:
-	dh $@ --with ocaml
-
+	dh $@
+# --buildsystem=cmake --builddirectory=$(TARGET_BUILD)
 
 debian/%: debian/%.in
 	sed -e 's|@DEB_HOST_MULTIARCH@|$(DEB_HOST_MULTIARCH)|g' -e "s|@OCAML_STDLIB_DIR@|$(OCAML_STDLIB_DIR)|g" -e "s|@LLVM_VERSION@|$(LLVM_VERSION)|g" $< > $@
 
-
-override_dh_ocamlinit:
-override_dh_ocamlclean:
-
-
-override_dh_auto_configure: debian/lldb-$(LLVM_VERSION).install debian/lldb-$(LLVM_VERSION).links debian/libllvm$(LLVM_VERSION).install debian/llvm-$(LLVM_VERSION)-dev.links debian/libllvm-$(LLVM_VERSION)-ocaml-dev.install debian/libllvm-$(LLVM_VERSION)-ocaml-dev.dirs debian/llvm-$(LLVM_VERSION)-runtime.binfmt debian/llvm-$(LLVM_VERSION)-runtime.prerm debian/libclang1-$(LLVM_VERSION).links debian/libclang-$(LLVM_VERSION)-dev.links debian/libclang1-$(LLVM_VERSION).install debian/pollycc
+override_dh_auto_configure: debian/lldb-$(LLVM_VERSION).install debian/lldb-$(LLVM_VERSION).links debian/libllvm$(LLVM_VERSION).install debian/llvm-$(LLVM_VERSION)-dev.links debian/libllvm-$(LLVM_VERSION)-ocaml-dev.install debian/libllvm-$(LLVM_VERSION)-ocaml-dev.dirs debian/llvm-$(LLVM_VERSION)-runtime.binfmt debian/libclang1-$(LLVM_VERSION).links debian/libclang-$(LLVM_VERSION)-dev.links debian/libclang1-$(LLVM_VERSION).install debian/pollycc
 	echo "Using gcc: "
 	$(CC) -v
 	mkdir -p $(TARGET_BUILD)
@@ -176,11 +170,12 @@
 		cd $$d && rm -f underscore.js && ln -s /usr/share/javascript/underscore/underscore.js; \
 	done
 
-	help2man --version-string=$(LLVM_VERSION) clang/tools/scan-view/scan-view > debian/man/scan-view.1
+	cp clang/tools/scan-build/scan-build.1 debian/man/scan-build-$(LLVM_VERSION).1
+	help2man --version-string=$(LLVM_VERSION) clang/tools/scan-view/scan-view > debian/man/scan-view-$(LLVM_VERSION).1
 	help2man --version-string=$(LLVM_VERSION) clang/tools/clang-format/clang-format-diff.py > debian/man/clang-format-diff-$(LLVM_VERSION).1
 
 	for f in clang clang-tblgen clang-check; do \
-		help2man --version-string=$(LLVM_VERSION) $(TARGET_BUILD)/Release/bin/$$f > debian/man/$$f.1; \
+		help2man --version-string=$(LLVM_VERSION) $(TARGET_BUILD)/Release/bin/$$f > debian/man/$$f-$(LLVM_VERSION).1; \
 	done
 
 	for f in llvm-dwarfdump llvm-mc llvm-mcmarkup llvm-objdump llvm-rtdyld llvm-size llvm-tblgen macho-dump lldb clang-format cpp11-migrate; do \
@@ -207,8 +202,8 @@
 
 #	cd $(TARGET_BUILD)/tools/clang/runtime/compiler-rt/ && rm -rf $$(find . -iname "SubDir.lib*" -o -iname .dir) && rm -rf $$(find -empty)
 
-	mkdir -p $(CURDIR)/debian/clang-$(LLVM_VERSION)/usr/bin/
-	cp compiler-rt/lib/asan/scripts/asan_symbolize.py $(CURDIR)/debian/clang-$(LLVM_VERSION)/usr/bin/asan_symbolize
+	mkdir -p $(CURDIR)/debian/clang-$(LLVM_VERSION)/usr/lib/llvm-$(LLVM_VERSION)/bin
+	cp compiler-rt/lib/asan/scripts/asan_symbolize.py $(CURDIR)/debian/clang-$(LLVM_VERSION)/usr/lib/llvm-$(LLVM_VERSION)/bin/asan_symbolize
 
 # Create this fake directory to make the install libclang-common-dev happy
 # under the unsupported archs of compiler-rt
@@ -227,14 +222,6 @@
 # Rename some stuff with the version name
 	mv $(CURDIR)/$(TARGET_BUILD)/tools/clang/docs/tools/clang.1 $(CURDIR)/$(TARGET_BUILD)/tools/clang/docs/tools/clang-$(LLVM_VERSION).1
 
-	cp -f $(CURDIR)/utils/vim/llvm.vim $(CURDIR)/utils/vim/llvm-$(LLVM_VERSION).vim
-
-	cp -f $(CURDIR)/utils/vim/tablegen.vim $(CURDIR)/utils/vim/tablegen-$(LLVM_VERSION).vim
-
-	cp -f $(CURDIR)/clang/tools/clang-format/clang-format-diff.py $(CURDIR)/clang/tools/clang-format/clang-format-diff-$(LLVM_VERSION)
-
-	cp -f $(CURDIR)/clang/tools/clang-format/clang-format.py clang/tools/clang-format/clang-format-$(LLVM_VERSION).py
-
 # Remove some license files
 	rm -f $(CURDIR)/debian/tmp/usr/lib/llvm-$(LLVM_VERSION)/include/llvm/Support/LICENSE.TXT \
 	$(CURDIR)/debian/llvm-$(LLVM_VERSION)-dev/usr/lib/llvm-$(LLVM_VERSION)/build/autoconf/LICENSE.TXT
@@ -242,6 +229,9 @@
 # Managed in lldb-X.Y.links.in
 	rm -f $(CURDIR)/$(TARGET_BUILD)/Release/lib/python*/site-packages/lldb/_lldb.so
 
+	cp $(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev.META \
+		$(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev/$(OCAML_STDLIB_DIR)/METAS/META.llvm-$(subst .,_,$(LLVM_VERSION))
+
 override_dh_installman:
 	dh_installman
 # Make sure that lli manpage is only in llvm-3.2-runtime (See #697117)
@@ -250,8 +240,6 @@
 
 .PHONY: override_dh_strip
 override_dh_strip:
-	dh_strip -p libclang$(SONAME_EXT)-$(LLVM_VERSION) --dbg-package=libclang$(SONAME_EXT)-$(LLVM_VERSION)-dbg
-	dh_strip -p libllvm$(LLVM_VERSION) --dbg-package=libllvm$(LLVM_VERSION)-dbg
 	dh_strip -a
 
 override_dh_shlibdeps:
@@ -264,7 +252,14 @@
 	rm -f $(CURDIR)/debian/lldb-$(LLVM_VERSION)/usr/lib/python*/dist-packages/lldb/__init__.pyc
 	rm -f $(CURDIR)/debian/clang-$(LLVM_VERSION)-examples/usr/share/doc/clang-$(LLVM_VERSION)-examples/examples/*Make*
 
+	find $(CURDIR)/debian/python-clang-$(LLVM_VERSION)/usr/share/doc/python-clang-$(LLVM_VERSION)/examples -name "*.gz" -exec gzip -d {} \;
+	find $(CURDIR)/debian/python-clang-$(LLVM_VERSION)/usr/share/doc/python-clang-$(LLVM_VERSION)/tests    -name "*.gz" -exec gzip -d {} \;
+	if test -d $(CURDIR)/debian/clang-$(LLVM_VERSION)-relaxng/usr/share/xml/clang-$(LLVM_VERSION)-relaxng ; then \
+		find $(CURDIR)/debian/clang-$(LLVM_VERSION)-relaxng/usr/share/xml/clang-$(LLVM_VERSION)-relaxng -name "*.gz" -exec gzip -d {} \; ; \
+	fi
 
+	chmod +x $(CURDIR)/debian/clang-$(LLVM_VERSION)/usr/lib/llvm-3.3/bin/pollycc
+	chmod +x $(CURDIR)/debian/clang-format-$(LLVM_VERSION)/usr/lib/llvm-3.3/bin/clang-format.py
 
 ifeq (,$(filter nocheck, $(DEB_BUILD_OPTIONS)))
 override_dh_auto_test:
@@ -314,7 +309,12 @@
 	find test/Bindings -name '*.o' | xargs -r rm -f
 	rm -f tools/clang tools/polly tools/lldb projects/compiler-rt
 	rm -rf tools/clang/tools/extra clang/tools/extra/
-	rm -f $(CURDIR)/utils/vim/llvm-$(LLVM_VERSION).vim $(CURDIR)/utils/vim/tablegen-$(LLVM_VERSION).vim
-	rm -f $(CURDIR)/clang/tools/clang-format/clang-format-diff-$(LLVM_VERSION)
-	rm -f $(CURDIR)/clang/tools/clang-format/clang-format-$(LLVM_VERSION).py
+	rm -f debian/man/scan-build-$(LLVM_VERSION).1
+	rm -f debian/man/scan-view-$(LLVM_VERSION).1
+	for f in clang clang-tblgen clang-check; do \
+		rm -f debian/man/$$f-$(LLVM_VERSION).1; \
+	done
+	for f in llvm-dwarfdump llvm-mc llvm-mcmarkup llvm-objdump llvm-rtdyld llvm-size llvm-tblgen macho-dump lldb clang-format cpp11-migrate; do \
+		rm -f debian/man/$$f-$(LLVM_VERSION).1; \
+	done
 
