Index: llvm-3.2.src/lldb/source/Symbol/ClangASTContext.cpp
===================================================================
--- llvm-3.2.src.orig/lldb/source/Symbol/ClangASTContext.cpp	2013-01-10 17:51:58.000000000 +0100
+++ llvm-3.2.src/lldb/source/Symbol/ClangASTContext.cpp	2013-01-12 15:38:43.799057785 +0100
@@ -257,7 +257,7 @@
     }
 
     const LangStandard &Std = LangStandard::getLangStandardForKind(LangStd);
-    Opts.BCPLComment = Std.hasBCPLComments();
+    Opts.LineComment = Std.hasLineComments();
     Opts.C99 = Std.isC99();
     Opts.CPlusPlus = Std.isCPlusPlus();
     Opts.CPlusPlus0x = Std.isCPlusPlus0x();
