Index: lldb-3.2/lldb/source/Core/cxa_demangle.cpp
===================================================================
--- lldb-3.2.orig/lldb/source/Core/cxa_demangle.cpp	(revision 168834)
+++ lldb-3.2/lldb/source/Core/cxa_demangle.cpp	(revision 168835)
@@ -3932,11 +3932,11 @@
     }
 };
 
-class __lambda
+class __lambda_node
     : public __node
 {
 public:
-    __lambda(__node* params, const char *number, size_t number_size)
+    __lambda_node(__node* params, const char *number, size_t number_size)
     {
         __right_ = params;
         __name_ = number;
@@ -7009,7 +7009,7 @@
                 return first;
               
             if (type == 'l')
-                __make<__lambda>(__root_, number_start, static_cast<size_t>(number_end - number_start));
+                __make<__lambda_node>(__root_, number_start, static_cast<size_t>(number_end - number_start));
             else
                 __make<__unnamed>(number_start, static_cast<size_t>(number_end - number_start));
             
