Index: llvm-toolchain-snapshot_3.5~svn211089/lldb/source/Expression/ClangExpressionParser.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn211089.orig/lldb/source/Expression/ClangExpressionParser.cpp	(révision 215101)
+++ llvm-toolchain-snapshot_3.5~svn211089/lldb/source/Expression/ClangExpressionParser.cpp	(copie de travail)
@@ -231,9 +231,11 @@
 
     // Disable some warnings.
     m_compiler->getDiagnostics().setSeverityForGroup(
-        "unused-value", clang::diag::Severity::Ignored, SourceLocation());
+        diag::Flavor::Remark, "unused-value", clang::diag::Severity::Ignored,
+        SourceLocation());
     m_compiler->getDiagnostics().setSeverityForGroup(
-        "odr", clang::diag::Severity::Ignored, SourceLocation());
+        diag::Flavor::Remark, "odr", clang::diag::Severity::Ignored,
+        SourceLocation());
 
     // Inform the target of the language options
     //
