Index: llvm-toolchain-snapshot_3.5~svn199197/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.5~svn199197.orig/clang/tools/scan-build/scan-build	2014-01-14 11:30:50.852868672 +0100
+++ llvm-toolchain-snapshot_3.5~svn199197/clang/tools/scan-build/scan-build	2014-01-14 11:30:50.848868672 +0100
@@ -1559,7 +1559,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/lib/llvm-3.5/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
diff -Naur llvm-toolchain-3.5-3.5~+rc1.orig/clang/tools/scan-build/ccc-analyzer llvm-toolchain-3.5-3.5~+rc1/clang/tools/scan-build/ccc-analyzer
--- llvm-toolchain-3.5-3.5~+rc1.orig/clang/tools/scan-build/ccc-analyzer	2014-08-06 10:29:51.990552103 +0200
+++ llvm-toolchain-3.5-3.5~+rc1/clang/tools/scan-build/ccc-analyzer	2014-08-06 10:35:05.016567533 +0200
@@ -25,6 +25,17 @@
 # Compiler command setup.
 ##===----------------------------------------------------------------------===##
 
+# Search in the PATH if the compiler exists
+sub SearchInPath {
+    my $file = shift;
+    foreach my $dir (split (':', $ENV{PATH})) {
+        if (-x "$dir/$file") {
+            return 1;
+        }
+    }
+    return 0;
+}
+
 my $Compiler;
 my $Clang;
 my $DefaultCCompiler;
@@ -41,7 +52,7 @@
 
 if ($FindBin::Script =~ /c\+\+-analyzer/) {
   $Compiler = $ENV{'CCC_CXX'};
-  if (!defined $Compiler || ! -x $Compiler) { $Compiler = $DefaultCXXCompiler; }
+  if (!defined $Compiler || (! -x $Compiler && ! SearchInPath($Compiler))) { $Compiler = $DefaultCXXCompiler; }
 
   $Clang = $ENV{'CLANG_CXX'};
   if (!defined $Clang || ! -x $Clang) { $Clang = 'clang++'; }
@@ -50,7 +61,7 @@
 }
 else {
   $Compiler = $ENV{'CCC_CC'};
-  if (!defined $Compiler || ! -x $Compiler) { $Compiler = $DefaultCCompiler; }
+  if (!defined $Compiler || (! -x $Compiler && ! SearchInPath($Compiler))) { $Compiler = $DefaultCCompiler; }
 
   $Clang = $ENV{'CLANG'};
   if (!defined $Clang || ! -x $Clang) { $Clang = 'clang'; }
