Index: llvm-toolchain-3.3-3.3/lldb/include/lldb/Makefile
===================================================================
--- llvm-toolchain-3.3-3.3.orig/lldb/include/lldb/Makefile	(révision 191067)
+++ llvm-toolchain-3.3-3.3/lldb/include/lldb/Makefile	(copie de travail)
@@ -4,11 +4,11 @@
 include $(LEVEL)/Makefile.common
 
 install-local::
-	$(Echo) Installing Clang include files
+	$(Echo) Installing LLDB include files
 	$(Verb) $(MKDIR) $(DESTDIR)$(PROJ_includedir)
-	$(Verb) if test -d "$(PROJ_SRC_ROOT)/tools/clang/include/clang" ; then \
-	  cd $(PROJ_SRC_ROOT)/tools/clang/include && \
-	  for  hdr in `find clang -type f '!' '(' -name '*~' \
+	$(Verb) if test -d "$(PROJ_SRC_ROOT)/tools/lldb/include/lldb" ; then \
+	  cd $(PROJ_SRC_ROOT)/tools/lldb/include && \
+	  for  hdr in `find lldb -type f '!' '(' -name '*~' \
 	      -o -name '.#*' -o -name '*.in' -o -name '*.txt' \
 	      -o -name 'Makefile' -o -name '*.td' -o -name '*.orig' ')' -print \
               | grep -v CVS | grep -v .svn | grep -v .dir` ; do \
@@ -21,9 +21,9 @@
 	  done ; \
 	fi
 ifneq ($(PROJ_SRC_ROOT),$(PROJ_OBJ_ROOT))
-	$(Verb) if test -d "$(PROJ_OBJ_ROOT)/tools/clang/include/clang" ; then \
-	  cd $(PROJ_OBJ_ROOT)/tools/clang/include && \
-	  for hdr in `find clang -type f '!' '(' -name 'Makefile' ')' -print \
+	$(Verb) if test -d "$(PROJ_OBJ_ROOT)/tools/lldb/include/lldb" ; then \
+	  cd $(PROJ_OBJ_ROOT)/tools/lldb/include && \
+	  for hdr in `find lldb -type f '!' '(' -name 'Makefile' ')' -print \
             | grep -v CVS | grep -v .tmp | grep -v .dir` ; do \
 	    $(DataInstall) $$hdr $(DESTDIR)$(PROJ_includedir)/$$hdr ; \
 	  done ; \
