diff -urN debian/changelog clang-3.1/debian/changelog
--- debian/changelog	2012-06-28 07:10:09.000000000 +0900
+++ clang-3.1/debian/changelog	2013-06-22 03:51:47.124834472 +0900
@@ -1,3 +1,9 @@
+clang-3.1 (3.1-debian8ppa3) precise; urgency=low
+
+  * ppa for precise
+
+ -- ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>  Sat, 22 Jun 2013 03:45:00 +0900
+
 clang (3.1-8) unstable; urgency=low
 
   * Also ship clang static libraries for the plugin developments
diff -urN debian/control clang-3.1/debian/control
--- debian/control	2012-06-28 07:12:08.000000000 +0900
+++ clang-3.1/debian/control	2013-06-23 23:23:19.271434882 +0900
@@ -1,9 +1,7 @@
-Source: clang
+Source: clang-3.1
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Arthur Loiret <aloiret@debian.org>,
- Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 6.0.0), flex, bison, dejagnu, tcl8.5, expect,
     autoconf, automake1.9, perl, libtool, doxygen, chrpath, texinfo,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
@@ -11,16 +9,14 @@
     gcc-4.6, g++-4.6
 Build-Conflicts: oprofile, ocaml
 Standards-Version: 3.9.3
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/clang/trunk/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/clang/trunk/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 
-Package: clang
+Package: clang-3.1
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang-common-dev (= ${binary:Version})
-Provides: clang-3.1, c-compiler
+ libclang-3.1-common-dev (= ${binary:Version})
 Recommends: llvm-3.1-dev, python
 Description: Low-Level Virtual Machine (LLVM), C language family frontend
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
@@ -58,7 +54,7 @@
     * A single unified parser for C, Objective C, C++, and Objective C++
     * Conformance with C/C++/ObjC and their variants
 
-Package: libclang1
+Package: libclang1-3.1
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev
 Description: clang library
@@ -99,11 +95,11 @@
  .
  This package contains the clang library.
 
-Package: libclang-dev
+Package: libclang-3.1-dev
 Architecture: any
 Section: libdevel
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang1 (= ${binary:Version}), libclang-common-dev
+ libclang1-3.1 (= ${binary:Version}), libclang-3.1-common-dev
 Description: clang library - Development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
  tools that make it easy to build compilers, optimizers, Just-In-Time
@@ -142,9 +138,9 @@
  .
  This package contains the clang headers to develop extensions over libclang.
 
-Package: libclang-common-dev
+Package: libclang-3.1-common-dev
 Architecture: any
-Replaces: libclang-dev
+Replaces: libclang-3.1-dev
 Section: libdevel
 Description: clang library - Common development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
@@ -183,3 +179,46 @@
     * Conformance with C/C++/ObjC and their variants
  .
  This package contains the clang generic headers.
+
+Package: python-clang-3.1
+Architecture: any
+Section: python
+Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
+ libclang1-3.1 (= ${binary:Version}), python2.7
+Description: clang python bindings
+ The Low-Level Virtual Machine (LLVM) is a collection of libraries and
+ tools that make it easy to build compilers, optimizers, Just-In-Time
+ code generators, and many other compiler-related programs. LLVM
+ uses a single, language-independent virtual instruction set both
+ as an offline code representation (to communicate code between
+ compiler phases and to run-time systems) and as the compiler internal
+ representation (to analyze and transform programs). This persistent
+ code representation allows a common set of sophisticated compiler
+ techniques to be applied at compile-time, link-time, install-time,
+ run-time, or "idle-time" (between program runs).
+ .
+ Clang project is a C, C++, Objective C and Objective C++ front-end
+ for the LLVM compiler. Some of its goals include the following:
+ .
+ End-User Features:
+ .
+    * Fast compiles and low memory use
+    * Expressive diagnostics (examples)
+    * GCC compatibility
+ .
+ Utility and Applications:
+ .
+    * Modular library based architecture
+    * Support diverse clients (refactoring, static analysis, code
+      generation, etc)
+    * Allow tight integration with IDEs
+    * Use the LLVM 'BSD' License
+ .
+ Internal Design and Implementation:
+ .
+    * A real-world, production quality compiler
+    * A simple and hackable code base
+    * A single unified parser for C, Objective C, C++, and Objective C++
+    * Conformance with C/C++/ObjC and their variants
+ .
+ This package contains the clang python bindings.
diff -urN debian/control.in/clang clang-3.1/debian/control.in/clang
--- debian/control.in/clang	2012-06-21 18:03:51.000000000 +0900
+++ clang-3.1/debian/control.in/clang	2013-01-11 00:14:37.991880345 +0900
@@ -1,8 +1,7 @@
-Package: clang
+Package: clang-@UVERSION@
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang-common-dev (= ${binary:Version})
-Provides: clang-@UVERSION@, c-compiler
+ libclang-@UVERSION@-common-dev (= ${binary:Version})
 Recommends: llvm-@UVERSION@-dev, python
 Description: Low-Level Virtual Machine (LLVM), C language family frontend
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
diff -urN debian/control.in/libclang-common-dev clang-3.1/debian/control.in/libclang-common-dev
--- debian/control.in/libclang-common-dev	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/control.in/libclang-common-dev	2013-01-10 22:47:46.709003487 +0900
@@ -1,7 +1,7 @@
 
-Package: libclang-common-dev
+Package: libclang-@UVERSION@-common-dev
 Architecture: any
-Replaces: libclang-dev
+Replaces: libclang-@UVERSION@-dev
 Section: libdevel
 Description: clang library - Common development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
diff -urN debian/control.in/libclang-dev clang-3.1/debian/control.in/libclang-dev
--- debian/control.in/libclang-dev	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/control.in/libclang-dev	2013-01-10 22:48:25.725196953 +0900
@@ -1,9 +1,9 @@
 
-Package: libclang-dev
+Package: libclang-@UVERSION@-dev
 Architecture: any
 Section: libdevel
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang1 (= ${binary:Version}), libclang-common-dev
+ libclang1-@UVERSION@ (= ${binary:Version}), libclang-@UVERSION@-common-dev
 Description: clang library - Development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
  tools that make it easy to build compilers, optimizers, Just-In-Time
diff -urN debian/control.in/libclang1 clang-3.1/debian/control.in/libclang1
--- debian/control.in/libclang1	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/control.in/libclang1	2013-01-10 22:48:44.461289869 +0900
@@ -1,4 +1,4 @@
-Package: libclang1
+Package: libclang1-@UVERSION@
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev
 Description: clang library
diff -urN debian/control.in/python-clang clang-3.1/debian/control.in/python-clang
--- debian/control.in/python-clang	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.1/debian/control.in/python-clang	2013-06-23 12:32:49.882876731 +0900
@@ -0,0 +1,43 @@
+
+Package: python-clang-@UVERSION@
+Architecture: any
+Section: python
+Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
+ libclang1-@UVERSION@ (= ${binary:Version}), python2.7
+Description: clang python bindings
+ The Low-Level Virtual Machine (LLVM) is a collection of libraries and
+ tools that make it easy to build compilers, optimizers, Just-In-Time
+ code generators, and many other compiler-related programs. LLVM
+ uses a single, language-independent virtual instruction set both
+ as an offline code representation (to communicate code between
+ compiler phases and to run-time systems) and as the compiler internal
+ representation (to analyze and transform programs). This persistent
+ code representation allows a common set of sophisticated compiler
+ techniques to be applied at compile-time, link-time, install-time,
+ run-time, or "idle-time" (between program runs).
+ .
+ Clang project is a C, C++, Objective C and Objective C++ front-end
+ for the LLVM compiler. Some of its goals include the following:
+ .
+ End-User Features:
+ .
+    * Fast compiles and low memory use
+    * Expressive diagnostics (examples)
+    * GCC compatibility
+ .
+ Utility and Applications:
+ .
+    * Modular library based architecture
+    * Support diverse clients (refactoring, static analysis, code
+      generation, etc)
+    * Allow tight integration with IDEs
+    * Use the LLVM 'BSD' License
+ .
+ Internal Design and Implementation:
+ .
+    * A real-world, production quality compiler
+    * A simple and hackable code base
+    * A single unified parser for C, Objective C, C++, and Objective C++
+    * Conformance with C/C++/ObjC and their variants
+ .
+ This package contains the clang python bindings.
diff -urN debian/control.in/source clang-3.1/debian/control.in/source
--- debian/control.in/source	2012-06-21 18:03:51.000000000 +0900
+++ clang-3.1/debian/control.in/source	2013-01-10 22:49:59.329661125 +0900
@@ -1,9 +1,7 @@
 Source: @PKGSOURCE@
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Arthur Loiret <aloiret@debian.org>,
- Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 6.0.0), flex, bison, dejagnu, tcl8.5, expect,
     autoconf, automake1.9, perl, libtool, doxygen, chrpath, texinfo,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
@@ -11,8 +9,7 @@
     gcc-4.6, g++-4.6
 Build-Conflicts: oprofile @BUILDCONFLICTS@
 Standards-Version: 3.9.3
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/clang/trunk/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/clang/trunk/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 
diff -urN debian/debhelper.in/clang.install clang-3.1/debian/debhelper.in/clang.install
--- debian/debhelper.in/clang.install	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/clang.install	2013-01-10 22:51:27.090096301 +0900
@@ -1,5 +1,4 @@
-debian/@TMP@/@PF@/bin/clang			@PF@/bin/
-debian/@TMP@/@PF@/bin/clang++			@PF@/bin/
-tools/clang/tools/scan-build			@PF@/share/clang/
-tools/clang/tools/scan-view			@PF@/share/clang/
-debian/@TMP@/@PF@/share/man/man1/clang.1	usr/share/man/man1/
+debian/@TMP@/@PF@/bin/clang			@PF@/lib/llvm-3.1/bin/
+debian/@TMP@/@PF@/bin/clang++			@PF@/lib/llvm-3.1/bin/
+tools/clang/tools/scan-build			@PF@/lib/llvm-3.1/share/
+tools/clang/tools/scan-view			@PF@/lib/llvm-3.1/share/
diff -urN debian/debhelper.in/clang.links clang-3.1/debian/debhelper.in/clang.links
--- debian/debhelper.in/clang.links	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/clang.links	2013-01-10 22:54:47.675090945 +0900
@@ -1,5 +1,7 @@
-usr/share/man/man1/clang.1.gz		usr/share/man/man1/llvm-clang.1.gz
-usr/share/man/man1/clang.1.gz		usr/share/man/man1/clang++.1.gz
-@PF@/share/clang/scan-build/scan-build	@PF@/bin/scan-build
-@PF@/share/clang/scan-view/scan-view	@PF@/bin/scan-view
-@PF@/bin/clang				@PF@/bin/llvm-clang
+@PF@/lib/llvm-3.1/bin/clang			@PF@/bin/llvm-clang-3.1
+@PF@/lib/llvm-3.1/bin/clang			@PF@/bin/clang-3.1
+@PF@/lib/llvm-3.1/bin/clang++			@PF@/bin/clang++-3.1
+@PF@/lib/llvm-3.1/share/scan-build/scan-build	@PF@/bin/scan-build-3.1
+@PF@/lib/llvm-3.1/share/scan-view/scan-view	@PF@/bin/scan-view-3.1
+@PF@/share/man/man1/clang-3.1.1.gz		@PF@/share/man/man1/llvm-clang-3.1.1.gz
+@PF@/share/man/man1/clang-3.1.1.gz		@PF@/share/man/man1/clang++-3.1.1.gz
diff -urN debian/debhelper.in/clang.manpages clang-3.1/debian/debhelper.in/clang.manpages
--- debian/debhelper.in/clang.manpages	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/clang.manpages	2013-01-10 22:55:26.803284984 +0900
@@ -1 +1,3 @@
-debian/man/scan*.1
+debian/man/scan-build.1
+debian/man/scan-view.1
+debian/@TMP@/@PF@/share/man/man1/clang.1
diff -urN debian/debhelper.in/clang.postinst clang-3.1/debian/debhelper.in/clang.postinst
--- debian/debhelper.in/clang.postinst	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/clang.postinst	1970-01-01 09:00:00.000000000 +0900
@@ -1,19 +0,0 @@
-#! /bin/sh -e
-
-prio=10
-
-update-alternatives --quiet \
-	--install /usr/bin/c++ \
-		c++ \
-		/usr/bin/clang++ \
-		$prio
-
-update-alternatives --quiet \
-	--install /usr/bin/cc \
-		cc \
-		/usr/bin/clang \
-		$prio
-
-#DEBHELPER#
-
-exit 0
diff -urN debian/debhelper.in/clang.prerm clang-3.1/debian/debhelper.in/clang.prerm
--- debian/debhelper.in/clang.prerm	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/clang.prerm	1970-01-01 09:00:00.000000000 +0900
@@ -1,12 +0,0 @@
-#!/bin/sh
-
-set -e
-
-if [ "$1" = "remove" ] || [ "$1" = "deconfigure" ]; then
-    update-alternatives --quiet --remove cc /usr/bin/clang
-    update-alternatives --quiet --remove c++ /usr/bin/clang++
-fi
-
-#DEBHELPER#
-
-exit 0
diff -urN debian/debhelper.in/libclang-dev.install clang-3.1/debian/debhelper.in/libclang-dev.install
--- debian/debhelper.in/libclang-dev.install	2012-06-28 07:10:09.000000000 +0900
+++ clang-3.1/debian/debhelper.in/libclang-dev.install	2013-01-10 23:02:10.957289064 +0900
@@ -1,4 +1,4 @@
-debian/@TMP@/@PF@/include/clang                  @PF@/include/
-debian/@TMP@/@PF@/include/clang-c                @PF@/include/
-debian/@TMP@/@PF@/lib/libclang.so                @PF@/lib/
-debian/@TMP@/@PF@/lib/libclang*a                 @PF@/lib/
+debian/@TMP@/@PF@/include/clang                  @PF@/lib/llvm-3.1/include/
+debian/@TMP@/@PF@/include/clang-c                @PF@/lib/llvm-3.1/include/
+debian/@TMP@/@PF@/lib/libclang.so                @PF@/lib/llvm-3.1/lib/
+debian/@TMP@/@PF@/lib/libclang*a                 @PF@/lib/llvm-3.1/lib/
diff -urN debian/debhelper.in/libclang1.install clang-3.1/debian/debhelper.in/libclang1.install
--- debian/debhelper.in/libclang1.install	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/debhelper.in/libclang1.install	2013-01-10 22:57:04.235768119 +0900
@@ -1 +1 @@
-debian/@TMP@/@PF@/lib/libclang.so.1                       @PF@/lib/
+debian/@TMP@/@PF@/lib/libclang.so.1                       @PF@/lib/llvm-3.1/lib/
diff -urN debian/debhelper.in/python-clang.install clang-3.1/debian/debhelper.in/python-clang.install
--- debian/debhelper.in/python-clang.install	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.1/debian/debhelper.in/python-clang.install	2013-06-24 02:00:20.217628666 +0900
@@ -0,0 +1,4 @@
+tools/clang/bindings/python/clang/*	@PF@/lib/python2.7/dist-packages/clang31/
+tools/clang/bindings/python/README.txt	@PF@/share/doc/python-clang-3.1/
+tools/clang/bindings/python/examples	@PF@/share/doc/python-clang-3.1/
+tools/clang/bindings/python/tests	@PF@/share/doc/python-clang-3.1/
diff -urN debian/patches/28-gcc-4.7-paths-clang-3.1.diff clang-3.1/debian/patches/28-gcc-4.7-paths-clang-3.1.diff
--- debian/patches/28-gcc-4.7-paths-clang-3.1.diff	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.1/debian/patches/28-gcc-4.7-paths-clang-3.1.diff	2013-01-16 15:09:20.650567944 +0900
@@ -0,0 +1,50 @@
+--- clang-3.1-3.1.orig/tools/clang/lib/Driver/ToolChains.cpp
++++ clang-3.1-3.1/tools/clang/lib/Driver/ToolChains.cpp
+@@ -34,6 +34,9 @@
+ #include <cstdlib> // ::getenv
+ 
+ #include "clang/Config/config.h" // for GCC_INSTALL_PREFIX
++#include "clang/Debian/debian_path.h"
++
++#include <iostream>
+ 
+ using namespace clang::driver;
+ using namespace clang::driver::toolchains;
+@@ -2287,16 +2290,29 @@ void Linux::AddClangCXXStdlibIncludeArgs
+   StringRef LibDir = GCCInstallation.getParentLibPath();
+   StringRef InstallDir = GCCInstallation.getInstallPath();
+   StringRef Version = GCCInstallation.getVersion();
+-  if (!addLibStdCXXIncludePaths(LibDir + "/../include/c++/" + Version,
+-                                (GCCInstallation.getTriple().str() +
+-                                 GCCInstallation.getMultiarchSuffix()),
+-                                DriverArgs, CC1Args)) {
++  StringRef TripleStr = GCCInstallation.getTriple().str();
++
++  const std::string IncludePathCandidates[] = {
++    LibDir.str() + "/../include/c++/" + Version.str(),
++    // Once more, Debian headers locations changed (since gcc version 4.7.2-5)
++    // See Debian bug tracker #693240
++    LibDir.str() + "/../include/"+ TripleStr.str() +"/c++/" + Version.str(),
++    LibDir.str() + "/../include/" + DEB_HOST_MULTIARCH_TRIPLET + "/c++/" + Version.str(),
+     // Gentoo is weird and places its headers inside the GCC install, so if the
+     // first attempt to find the headers fails, try this pattern.
+-    addLibStdCXXIncludePaths(InstallDir + "/include/g++-v4",
+-                             (GCCInstallation.getTriple().str() +
+-                              GCCInstallation.getMultiarchSuffix()),
+-                             DriverArgs, CC1Args);
++    InstallDir.str() + "/include/g++-v4",
++    // Android standalone toolchain has C++ headers in yet another place.
++    LibDir.str() + "/../" + TripleStr.str() + "/include/c++/" + Version.str(),
++    // Freescale SDK C++ headers are directly in <sysroot>/usr/include/c++,
++    // without a subdirectory corresponding to the gcc version.
++    LibDir.str() + "/../include/c++",
++  };
++
++  for (unsigned i = 0; i < llvm::array_lengthof(IncludePathCandidates); ++i) {
++      addLibStdCXXIncludePaths(IncludePathCandidates[i], (TripleStr +
++                GCCInstallation.getMultiarchSuffix()),
++                               DriverArgs, CC1Args);
++
+   }
+ }
+ 
diff -urN debian/patches/99-python-bindings.patch clang-3.1/debian/patches/99-python-bindings.patch
--- debian/patches/99-python-bindings.patch	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.1/debian/patches/99-python-bindings.patch	2013-06-23 12:42:24.265724950 +0900
@@ -0,0 +1,149 @@
+diff -urN clang-3.1.orig/tools/clang/bindings/python/README.txt clang-3.1/tools/clang/bindings/python/README.txt
+--- clang-3.1.orig/tools/clang/bindings/python/README.txt	2010-05-01 06:51:10.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/README.txt	2013-06-23 12:37:19.028211350 +0900
+@@ -4,12 +4,9 @@
+ 
+ This directory implements Python bindings for Clang.
+ 
+-You may need to alter LD_LIBRARY_PATH so that the Clang library can be
+-found. The unit tests are designed to be run with 'nosetests'. For example:
++The unit tests are designed to be run with 'nosetests'. For example:
+ --
+-$ env PYTHONPATH=$(echo ~/llvm/tools/clang/bindings/python/) \
+-      LD_LIBRARY_PATH=$(llvm-config --libdir) \
+-  nosetests -v
++$ nosetests -v
+ tests.cindex.test_index.test_create ... ok
+ ...
+ 
+diff -urN clang-3.1.orig/tools/clang/bindings/python/clang/cindex.py clang-3.1/tools/clang/bindings/python/clang/cindex.py
+--- clang-3.1.orig/tools/clang/bindings/python/clang/cindex.py	2012-04-16 03:51:10.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/clang/cindex.py	2013-06-23 12:41:27.673444311 +0900
+@@ -76,7 +76,7 @@
+     elif name == 'Windows':
+         return cdll.LoadLibrary('libclang.dll')
+     else:
+-        return cdll.LoadLibrary('libclang.so')
++        return cdll.LoadLibrary('/usr/lib/llvm-3.1/lib/libclang.so.1')
+ 
+ # ctypes doesn't implicitly convert c_void_p to the appropriate wrapper
+ # object. This is a problem, because it means that from_parameter will see an
+diff -urN clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-dump.py
+--- clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2010-02-14 03:33:28.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2013-06-23 12:37:48.156355790 +0900
+@@ -54,7 +54,7 @@
+              'children' : children }
+ 
+ def main():
+-    from clang.cindex import Index
++    from clang31.cindex import Index
+     from pprint import pprint
+ 
+     from optparse import OptionParser, OptionGroup
+diff -urN clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-includes.py
+--- clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2010-02-14 03:33:28.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2013-06-23 12:37:59.980414432 +0900
+@@ -16,7 +16,7 @@
+ 
+ def main():
+     import sys
+-    from clang.cindex import Index
++    from clang31.cindex import Index
+ 
+     from optparse import OptionParser, OptionGroup
+ 
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py	2012-03-11 07:23:27.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor.py	2013-06-23 12:39:15.172787294 +0900
+@@ -1,5 +1,5 @@
+-from clang.cindex import CursorKind
+-from clang.cindex import TypeKind
++from clang31.cindex import CursorKind
++from clang31.cindex import TypeKind
+ from .util import get_cursor
+ from .util import get_tu
+ 
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2012-02-05 20:42:09.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2013-06-23 12:38:14.772487787 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import CursorKind
++from clang31.cindex import CursorKind
+ 
+ def test_name():
+     assert CursorKind.UNEXPOSED_DECL.name is 'UNEXPOSED_DECL'
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_diagnostics.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2012-03-11 07:19:05.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2013-06-23 12:39:03.356728676 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang31.cindex import *
+ from .util import get_tu
+ 
+ # FIXME: We need support for invalid translation units to test better.
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_file.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_file.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_file.py	2011-10-31 09:07:19.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_file.py	2013-06-23 12:39:27.436848105 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import Index, File
++from clang31.cindex import Index, File
+ 
+ def test_file():
+   index = Index.create()
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_index.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_index.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_index.py	2010-01-24 11:02:07.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_index.py	2013-06-23 12:38:51.524670030 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang31.cindex import *
+ import os
+ 
+ kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_location.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_location.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_location.py	2012-03-11 07:19:05.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_location.py	2013-06-23 12:38:28.228554488 +0900
+@@ -1,7 +1,7 @@
+-from clang.cindex import Cursor
+-from clang.cindex import File
+-from clang.cindex import SourceLocation
+-from clang.cindex import SourceRange
++from clang31.cindex import Cursor
++from clang31.cindex import File
++from clang31.cindex import SourceLocation
++from clang31.cindex import SourceRange
+ from .util import get_cursor
+ from .util import get_tu
+ 
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_translation_unit.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2011-07-06 12:00:34.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-06-23 12:39:50.340961675 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang31.cindex import *
+ import os
+ 
+ kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_type.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_type.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_type.py	2012-04-16 03:51:10.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_type.py	2013-06-23 12:39:38.620903555 +0900
+@@ -1,5 +1,5 @@
+-from clang.cindex import CursorKind
+-from clang.cindex import TypeKind
++from clang31.cindex import CursorKind
++from clang31.cindex import TypeKind
+ from nose.tools import raises
+ from .util import get_cursor
+ from .util import get_tu
+diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/util.py clang-3.1/tools/clang/bindings/python/tests/cindex/util.py
+--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/util.py	2012-03-11 07:19:05.000000000 +0900
++++ clang-3.1/tools/clang/bindings/python/tests/cindex/util.py	2013-06-23 12:38:41.956622576 +0900
+@@ -1,7 +1,7 @@
+ # This file provides common utility functions for the test suite.
+ 
+-from clang.cindex import Cursor
+-from clang.cindex import Index
++from clang31.cindex import Cursor
++from clang31.cindex import Index
+ 
+ def get_tu(source, lang='c', all_warnings=False):
+     """Obtain a translation unit from source and language.
diff -urN debian/patches/series clang-3.1/debian/patches/series
--- debian/patches/series	2012-06-26 06:31:47.000000000 +0900
+++ clang-3.1/debian/patches/series	2013-06-23 12:43:26.182113889 +0900
@@ -4,3 +4,5 @@
 21-searchPathHeaders.diff
 23-strlcpy_strlcat_warning_removed.diff
 26-disable_SplitsOnSpaces.diff
+28-gcc-4.7-paths-clang-3.1.diff
+99-python-bindings.patch
diff -urN debian/rules clang-3.1/debian/rules
--- debian/rules	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/rules	2013-06-21 23:02:32.197853431 +0900
@@ -17,7 +17,7 @@
 
 define pkgname
   $(shell cat $(D)/debian/control.in/$1 | awk '/^Package: / {print $$2}' | \
-  sed -e 's/@UVERSION@/$(UVERSION)/;s/@GCC_VERSION@/$(gcc_version)/;s/@PKG_VERSION@/$(pkg_version)/')
+  sed -e 's/@UVERSION@/$(UVERSION)/;s/@GCC_VERSION@/$(GCC_VERSION)/;s/@PKG_VERSION@/$(pkg_version)/')
 endef
 
 define buildof
diff -urN debian/rules.d/binary.mk clang-3.1/debian/rules.d/binary.mk
--- debian/rules.d/binary.mk	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/rules.d/binary.mk	2013-06-23 23:45:16.010961739 +0900
@@ -73,6 +73,10 @@
 	dh_fixperms -p$(call pkgname,$*)
 	DH_VERBOSE=1 dh_makeshlibs -p$(call pkgname,$*)
 	DH_VERBOSE=1 dh_shlibdeps -p$(call pkgname,$*)
+	if test $(call pkgname,$*) = python-clang-$(UVERSION) ; then \
+		find $(D)/debian/python-clang-$(UVERSION)/usr/share/doc/python-clang-$(UVERSION)/examples -name "*.gz" -exec gzip -d {} \; ; \
+		find $(D)/debian/python-clang-$(UVERSION)/usr/share/doc/python-clang-$(UVERSION)/tests    -name "*.gz" -exec gzip -d {} \; ; \
+	fi
 	dh_installdeb -p$(call pkgname,$*)
 	dh_gencontrol -p$(call pkgname,$*)
 	dh_md5sums -p$(call pkgname,$*)
diff -urN debian/rules.d/build.mk clang-3.1/debian/rules.d/build.mk
--- debian/rules.d/build.mk	2012-06-26 06:31:47.000000000 +0900
+++ clang-3.1/debian/rules.d/build.mk	2013-01-10 23:36:20.912394102 +0900
@@ -32,6 +32,7 @@
 	dh_clean
 	find $(srcdir)/test/Bindings/Ocaml -type f \
 	  -name "*.o" -or -name "*.cmx" | xargs $(RM)
+	$(RM) -r tools/clang/include/clang/Debian
 	$(RM) -r $(D)/debian/tmp-*
 	$(RM) $(D)/test/Bindings/Ocaml/*.cm[io]
 	$(RM) $(D)/debian/log-check-* $(D)/debian/logwatch-*.pid
diff -urN debian/rules.d/control.mk clang-3.1/debian/rules.d/control.mk
--- debian/rules.d/control.mk	2012-06-01 21:50:18.000000000 +0900
+++ clang-3.1/debian/rules.d/control.mk	2013-06-23 12:44:00.011187558 +0900
@@ -1,7 +1,7 @@
 $(foreach var,BUILDDEPS BUILDCONFLICTS,$(eval \
   $(if $(strip $($(var))),$(eval $(var) :=, $($(var))),$(eval $(var) :=))))
 
-extra_packages:=libclang1 libclang-dev libclang-common-dev
+extra_packages:=libclang1 libclang-dev libclang-common-dev python-clang
 packages := $(foreach build,$(builds),$($(build)_packages))
 packages += $(extra_packages) 
 
@@ -11,7 +11,7 @@
 	    -e "s/ *@BUILDDEPS@/$(BUILDDEPS)/g"			\
 	    -e "s/ *@BUILDCONFLICTS@/$(BUILDCONFLICTS)/g"	\
 	    -e "s/@UVERSION@/$(UVERSION)/g"			\
-	    -e "s/@GCC_VERSION@/$(gcc_version)/g"		\
+	    -e "s/@GCC_VERSION@/$(GCC_VERSION)/g"		\
 	    -e "s/@PKG_VERSION@/$(pkg_version)/g"		\
 	    -e "s/ , /, /g"					\
 	    $@.in/source $(addprefix $@.in/,$(packages)) > $@
