Index: clang-2.9/tools/clang/Makefile
===================================================================
--- clang-2.9.orig/tools/clang/Makefile	2011-09-26 17:31:02.000000000 +0200
+++ clang-2.9/tools/clang/Makefile	2011-09-26 17:31:17.000000000 +0200
@@ -43,9 +43,7 @@
 
 # Set common Clang build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(CLANG_LEVEL)/include -I$(PROJ_OBJ_DIR)/$(CLANG_LEVEL)/include
-ifdef CLANG_VENDOR
-CPP.Flags += -DCLANG_VENDOR='"$(CLANG_VENDOR) "'
-endif
+CPP.Flags += -DCLANG_VENDOR='"Debian "'
 
 # Disable -fstrict-aliasing. Darwin disables it by default (and LLVM doesn't
 # work with it enabled with GCC), Clang/llvm-gcc don't support it yet, and newer
