diff -urN ../../clang-3.0/debian/changelog clang-3.0/debian/changelog
--- ../../clang-3.0/debian/changelog	2013-03-29 03:03:16.000000000 +0900
+++ clang-3.0/debian/changelog	2014-09-22 18:27:10.625000579 +0900
@@ -1,3 +1,9 @@
+clang-3.0 (1:3.0-debian6.2trusty1) trusty; urgency=low
+
+  * ppa for trusty
+
+ -- ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>  Mon, 22 Sep 2014 18:20:00 +0900
+
 clang (1:3.0-6.2) unstable; urgency=low
 
   * Non-maintainer upload with maintainer's approval.
diff -urN ../../clang-3.0/debian/control clang-3.0/debian/control
--- ../../clang-3.0/debian/control	2013-03-31 10:15:22.000000000 +0900
+++ clang-3.0/debian/control	2014-09-21 19:26:51.633753080 +0900
@@ -1,24 +1,22 @@
-Source: clang
+Source: clang-3.0
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Arthur Loiret <aloiret@debian.org>,
- Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 6.0.0), flex, bison, dejagnu, tcl8.5, expect,
     autoconf, automake1.9, perl, libtool, doxygen, chrpath, texinfo,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
-    lsb-release, patchutils, diffstat, xz-utils, llvm-3.0-dev (>= 3.0), llvm-3.0-source (>= 3.0)
-Build-Conflicts: oprofile, ocaml, ocaml-nox
+    lsb-release, patchutils, diffstat, xz-utils, llvm-3.0-dev (>= 3.0), llvm-3.0-source (>= 3.0), python,
+    gcc-4.6, g++-4.6
+Build-Conflicts: oprofile, ocaml
 Standards-Version: 3.9.2
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/clang/trunk/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/clang/trunk/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 
-Package: clang
+Package: clang-3.0
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang-common-dev
+ libclang-common-3.0-dev (= ${binary:Version})
 Recommends: llvm-3.0-dev, python
 Description: Low-Level Virtual Machine (LLVM), C language family frontend
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
@@ -56,7 +54,7 @@
     * A single unified parser for C, Objective C, C++, and Objective C++
     * Conformance with C/C++/ObjC and their variants
 
-Package: libclang1
+Package: libclang1-3.0
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev
 Description: clang library
@@ -97,11 +95,11 @@
  .
  This package contains the clang library.
 
-Package: libclang-dev
+Package: libclang-3.0-dev
 Architecture: any
 Section: libdevel
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang1 (= ${binary:Version}), libclang-common-dev
+ libclang1-3.0 (= ${binary:Version}), libclang-common-3.0-dev (= ${binary:Version})
 Description: clang library - Development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
  tools that make it easy to build compilers, optimizers, Just-In-Time
@@ -140,9 +138,9 @@
  .
  This package contains the clang headers to develop extensions over libclang.
 
-Package: libclang-common-dev
+Package: libclang-common-3.0-dev
 Architecture: any
-Replaces: libclang-dev
+Replaces: libclang-3.0-dev
 Section: libdevel
 Description: clang library - Common development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
@@ -181,3 +179,10 @@
     * Conformance with C/C++/ObjC and their variants
  .
  This package contains the clang generic headers.
+
+Package: python-clang-3.0
+Architecture: any
+Section: python
+Depends: ${shlibs:Depends}, ${misc:Depends}, libclang1-3.0 (= ${binary:Version}), python2.7
+Description: Clang Python Bindings
+ This binding package provides access to the Clang compiler and libraries.
diff -urN ../../clang-3.0/debian/control.in/clang clang-3.0/debian/control.in/clang
--- ../../clang-3.0/debian/control.in/clang	2011-12-20 22:40:25.000000000 +0900
+++ clang-3.0/debian/control.in/clang	2014-05-09 17:45:31.456192953 +0900
@@ -1,7 +1,7 @@
-Package: clang
+Package: clang-@UVERSION@
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang-common-dev
+ libclang-common-@UVERSION@-dev (= ${binary:Version})
 Recommends: llvm-@UVERSION@-dev, python
 Description: Low-Level Virtual Machine (LLVM), C language family frontend
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
diff -urN ../../clang-3.0/debian/control.in/libclang-common-dev clang-3.0/debian/control.in/libclang-common-dev
--- ../../clang-3.0/debian/control.in/libclang-common-dev	2011-12-26 08:50:52.000000000 +0900
+++ clang-3.0/debian/control.in/libclang-common-dev	2014-05-09 17:45:31.456192953 +0900
@@ -1,7 +1,7 @@
 
-Package: libclang-common-dev
+Package: libclang-common-@UVERSION@-dev
 Architecture: any
-Replaces: libclang-dev
+Replaces: libclang-@UVERSION@-dev
 Section: libdevel
 Description: clang library - Common development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
diff -urN ../../clang-3.0/debian/control.in/libclang-dev clang-3.0/debian/control.in/libclang-dev
--- ../../clang-3.0/debian/control.in/libclang-dev	2011-12-20 22:38:39.000000000 +0900
+++ clang-3.0/debian/control.in/libclang-dev	2014-05-09 17:45:31.456192953 +0900
@@ -1,9 +1,9 @@
 
-Package: libclang-dev
+Package: libclang-@UVERSION@-dev
 Architecture: any
 Section: libdevel
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev,
- libclang1 (= ${binary:Version}), libclang-common-dev
+ libclang1-@UVERSION@ (= ${binary:Version}), libclang-common-@UVERSION@-dev (= ${binary:Version})
 Description: clang library - Development package
  The Low-Level Virtual Machine (LLVM) is a collection of libraries and
  tools that make it easy to build compilers, optimizers, Just-In-Time
diff -urN ../../clang-3.0/debian/control.in/libclang1 clang-3.0/debian/control.in/libclang1
--- ../../clang-3.0/debian/control.in/libclang1	2011-10-08 01:14:48.000000000 +0900
+++ clang-3.0/debian/control.in/libclang1	2014-05-09 17:45:31.456192953 +0900
@@ -1,4 +1,4 @@
-Package: libclang1
+Package: libclang1-@UVERSION@
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}, libstdc++6-4.6-dev
 Description: clang library
diff -urN ../../clang-3.0/debian/control.in/python-clang clang-3.0/debian/control.in/python-clang
--- ../../clang-3.0/debian/control.in/python-clang	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/control.in/python-clang	2014-05-09 17:45:31.456192953 +0900
@@ -0,0 +1,7 @@
+
+Package: python-clang-@UVERSION@
+Architecture: any
+Section: python
+Depends: ${shlibs:Depends}, ${misc:Depends}, libclang1-@UVERSION@ (= ${binary:Version}), python2.7
+Description: Clang Python Bindings
+ This binding package provides access to the Clang compiler and libraries.
diff -urN ../../clang-3.0/debian/control.in/source clang-3.0/debian/control.in/source
--- ../../clang-3.0/debian/control.in/source	2011-12-08 00:39:17.000000000 +0900
+++ clang-3.0/debian/control.in/source	2014-07-26 09:18:32.987161133 +0900
@@ -1,17 +1,15 @@
 Source: @PKGSOURCE@
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Arthur Loiret <aloiret@debian.org>,
- Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 6.0.0), flex, bison, dejagnu, tcl8.5, expect,
     autoconf, automake1.9, perl, libtool, doxygen, chrpath, texinfo,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
-    lsb-release, patchutils, diffstat, xz-utils @BUILDDEPS@
+    lsb-release, patchutils, diffstat, xz-utils @BUILDDEPS@, python,
+    gcc-4.6, g++-4.6
 Build-Conflicts: oprofile @BUILDCONFLICTS@
 Standards-Version: 3.9.2
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/clang/trunk/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/clang/trunk/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 
diff -urN ../../clang-3.0/debian/debhelper.in/clang.install clang-3.0/debian/debhelper.in/clang.install
--- ../../clang-3.0/debian/debhelper.in/clang.install	2011-09-23 05:02:23.000000000 +0900
+++ clang-3.0/debian/debhelper.in/clang.install	2014-09-21 19:26:28.996963945 +0900
@@ -1,5 +1,4 @@
-debian/@TMP@/@PF@/bin/clang			@PF@/bin/
-debian/@TMP@/@PF@/bin/clang++			@PF@/bin/
-tools/clang/tools/scan-build			@PF@/share/clang/
-tools/clang/tools/scan-view			@PF@/share/clang/
-debian/@TMP@/@PF@/share/man/man1/clang.1	usr/share/man/man1/
+debian/@TMP@/@PF@/bin/clang			@PF@/lib/llvm-@UVERSION@/bin/
+debian/@TMP@/@PF@/bin/clang++			@PF@/lib/llvm-@UVERSION@/bin/
+tools/clang/tools/scan-build-@UVERSION@		usr/share/clang/
+tools/clang/tools/scan-view-@UVERSION@		usr/share/clang/
diff -urN ../../clang-3.0/debian/debhelper.in/clang.links clang-3.0/debian/debhelper.in/clang.links
--- ../../clang-3.0/debian/debhelper.in/clang.links	2011-06-27 17:12:38.000000000 +0900
+++ clang-3.0/debian/debhelper.in/clang.links	2014-09-21 19:26:35.785602125 +0900
@@ -1,5 +1,7 @@
-usr/share/man/man1/clang.1.gz		usr/share/man/man1/llvm-clang.1.gz
-usr/share/man/man1/clang.1.gz		usr/share/man/man1/clang++.1.gz
-@PF@/share/clang/scan-build/scan-build	@PF@/bin/scan-build
-@PF@/share/clang/scan-view/scan-view	@PF@/bin/scan-view
-@PF@/bin/clang				@PF@/bin/llvm-clang
+usr/share/man/man1/clang-@UVERSION@.1.gz		usr/share/man/man1/llvm-clang-@UVERSION@.1.gz
+usr/share/man/man1/clang-@UVERSION@.1.gz		usr/share/man/man1/clang++-@UVERSION@.1.gz
+usr/share/clang/scan-build-@UVERSION@/scan-build	usr/bin/scan-build-@UVERSION@
+usr/share/clang/scan-view-@UVERSION@/scan-view		usr/bin/scan-view-@UVERSION@
+usr/lib/llvm-@UVERSION@/bin/clang			usr/bin/llvm-clang-@UVERSION@
+usr/lib/llvm-@UVERSION@/bin/clang			usr/bin/clang-@UVERSION@
+usr/lib/llvm-@UVERSION@/bin/clang++			usr/bin/clang++-@UVERSION@
diff -urN ../../clang-3.0/debian/debhelper.in/clang.manpages clang-3.0/debian/debhelper.in/clang.manpages
--- ../../clang-3.0/debian/debhelper.in/clang.manpages	2011-06-27 17:12:38.000000000 +0900
+++ clang-3.0/debian/debhelper.in/clang.manpages	2014-09-21 00:02:03.644827851 +0900
@@ -1 +1,3 @@
-debian/man/scan*.1
+build-clang/tools/clang/docs/tools/clang.1
+debian/man/scan-build.1
+debian/man/scan-view.1
diff -urN ../../clang-3.0/debian/debhelper.in/clang.postinst clang-3.0/debian/debhelper.in/clang.postinst
--- ../../clang-3.0/debian/debhelper.in/clang.postinst	2011-06-27 17:12:38.000000000 +0900
+++ clang-3.0/debian/debhelper.in/clang.postinst	1970-01-01 09:00:00.000000000 +0900
@@ -1,19 +0,0 @@
-#! /bin/sh -e
-
-prio=10
-
-update-alternatives --quiet \
-	--install /usr/bin/c++ \
-		c++ \
-		/usr/bin/clang++ \
-		$prio
-
-update-alternatives --quiet \
-	--install /usr/bin/cc \
-		cc \
-		/usr/bin/clang \
-		$prio
-
-#DEBHELPER#
-
-exit 0
diff -urN ../../clang-3.0/debian/debhelper.in/clang.prerm clang-3.0/debian/debhelper.in/clang.prerm
--- ../../clang-3.0/debian/debhelper.in/clang.prerm	2011-06-27 17:12:38.000000000 +0900
+++ clang-3.0/debian/debhelper.in/clang.prerm	1970-01-01 09:00:00.000000000 +0900
@@ -1,12 +0,0 @@
-#!/bin/sh
-
-set -e
-
-if [ "$1" = "remove" ] || [ "$1" = "deconfigure" ]; then
-    update-alternatives --quiet --remove cc /usr/bin/clang
-    update-alternatives --quiet --remove c++ /usr/bin/clang++
-fi
-
-#DEBHELPER#
-
-exit 0
diff -urN ../../clang-3.0/debian/debhelper.in/libclang-dev.install clang-3.0/debian/debhelper.in/libclang-dev.install
--- ../../clang-3.0/debian/debhelper.in/libclang-dev.install	2011-12-20 22:41:09.000000000 +0900
+++ clang-3.0/debian/debhelper.in/libclang-dev.install	2014-09-20 23:36:33.534673113 +0900
@@ -1,3 +1,2 @@
-debian/@TMP@/@PF@/include/clang                  @PF@/include/
-debian/@TMP@/@PF@/include/clang-c                  @PF@/include/
-debian/@TMP@/@PF@/lib/libclang.so                       @PF@/lib/
+debian/@TMP@/@PF@/include/clang                  @PF@/lib/llvm-@UVERSION@/include/
+debian/@TMP@/@PF@/include/clang-c                @PF@/lib/llvm-@UVERSION@/include/
diff -urN ../../clang-3.0/debian/debhelper.in/libclang-dev.links clang-3.0/debian/debhelper.in/libclang-dev.links
--- ../../clang-3.0/debian/debhelper.in/libclang-dev.links	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/debhelper.in/libclang-dev.links	2014-09-20 23:37:05.638772591 +0900
@@ -0,0 +1,3 @@
+usr/lib/@DEB_HOST_MULTIARCH@/libclang-@UVERSION@.so.1 usr/lib/@DEB_HOST_MULTIARCH@/libclang-@UVERSION@.so
+usr/lib/@DEB_HOST_MULTIARCH@/libclang-@UVERSION@.so.1 usr/lib/llvm-@UVERSION@/lib/libclang-@UVERSION@.so
+usr/lib/@DEB_HOST_MULTIARCH@/libclang-@UVERSION@.so.1 usr/lib/llvm-@UVERSION@/lib/libclang.so
diff -urN ../../clang-3.0/debian/debhelper.in/libclang1.install clang-3.0/debian/debhelper.in/libclang1.install
--- ../../clang-3.0/debian/debhelper.in/libclang1.install	2011-10-08 01:14:24.000000000 +0900
+++ clang-3.0/debian/debhelper.in/libclang1.install	2014-09-20 23:37:19.555879046 +0900
@@ -1 +1 @@
-debian/@TMP@/@PF@/lib/libclang.so.1                       @PF@/lib/
+debian/@TMP@/@PF@/lib/libclang-@UVERSION@.so.1 /usr/lib/@DEB_HOST_MULTIARCH@/
diff -urN ../../clang-3.0/debian/debhelper.in/libclang1.links clang-3.0/debian/debhelper.in/libclang1.links
--- ../../clang-3.0/debian/debhelper.in/libclang1.links	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/debhelper.in/libclang1.links	2014-09-20 23:37:32.677380103 +0900
@@ -0,0 +1 @@
+usr/lib/@DEB_HOST_MULTIARCH@/libclang-@UVERSION@.so.1 usr/lib/llvm-@UVERSION@/lib/libclang.so.1
diff -urN ../../clang-3.0/debian/debhelper.in/python-clang.install clang-3.0/debian/debhelper.in/python-clang.install
--- ../../clang-3.0/debian/debhelper.in/python-clang.install	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/debhelper.in/python-clang.install	2014-09-20 23:37:48.049766606 +0900
@@ -0,0 +1,4 @@
+tools/clang/bindings/python/clang/*	@PF@/lib/python2.7/dist-packages/clang30/
+tools/clang/bindings/python/README.txt	@PF@/share/doc/python-clang-@UVERSION@/
+tools/clang/bindings/python/examples	@PF@/share/doc/python-clang-@UVERSION@/
+tools/clang/bindings/python/tests	@PF@/share/doc/python-clang-@UVERSION@/
diff -urN ../../clang-3.0/debian/packages.d/tools.mk clang-3.0/debian/packages.d/tools.mk
--- ../../clang-3.0/debian/packages.d/tools.mk	2011-12-07 18:08:40.000000000 +0900
+++ clang-3.0/debian/packages.d/tools.mk	2014-05-09 17:45:31.456192953 +0900
@@ -6,7 +6,7 @@
 
 BUILDDEPS	:= \
 	llvm-$(UVERSION)-dev (>= $(dev_version)), llvm-$(UVERSION)-source (>= $(dev_version))
-BUILDCONFLICTS	:= ocaml, ocaml-nox
+BUILDCONFLICTS	:= ocaml
 
 PF		:= /usr
 
diff -urN ../../clang-3.0/debian/patches/32-scan-build-path.diff clang-3.0/debian/patches/32-scan-build-path.diff
--- ../../clang-3.0/debian/patches/32-scan-build-path.diff	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/patches/32-scan-build-path.diff	2014-09-20 23:43:35.705578484 +0900
@@ -0,0 +1,11 @@
+--- clang-3.0-3.0.orig/tools/clang/tools/scan-build/scan-build
++++ clang-3.0-3.0/tools/clang/tools/scan-build/scan-build
+@@ -81,7 +81,7 @@ sub DieDiag {
+ ##----------------------------------------------------------------------------##
+ 
+ # Find 'clang'
+-my $ClangSB = Cwd::realpath("$RealBin/bin/clang");
++my $ClangSB = Cwd::realpath("/usr/lib/llvm-3.0/bin/clang");
+ if (!defined $ClangSB || ! -x $ClangSB) {
+   $ClangSB = Cwd::realpath("$RealBin/clang");
+ }
diff -urN ../../clang-3.0/debian/patches/99-python-bindings.patch clang-3.0/debian/patches/99-python-bindings.patch
--- ../../clang-3.0/debian/patches/99-python-bindings.patch	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/patches/99-python-bindings.patch	2014-05-09 17:45:31.456192953 +0900
@@ -0,0 +1,105 @@
+diff -urN clang-3.0.orig/tools/clang/bindings/python/README.txt clang-3.0/tools/clang/bindings/python/README.txt
+--- clang-3.0.orig/tools/clang/bindings/python/README.txt	2010-05-01 06:51:10.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/README.txt	2013-06-23 11:47:10.781294252 +0900
+@@ -4,12 +4,9 @@
+ 
+ This directory implements Python bindings for Clang.
+ 
+-You may need to alter LD_LIBRARY_PATH so that the Clang library can be
+-found. The unit tests are designed to be run with 'nosetests'. For example:
++The unit tests are designed to be run with 'nosetests'. For example:
+ --
+-$ env PYTHONPATH=$(echo ~/llvm/tools/clang/bindings/python/) \
+-      LD_LIBRARY_PATH=$(llvm-config --libdir) \
+-  nosetests -v
++$ nosetests -v
+ tests.cindex.test_index.test_create ... ok
+ ...
+ 
+diff -urN clang-3.0.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py clang-3.0/tools/clang/bindings/python/examples/cindex/cindex-dump.py
+--- clang-3.0.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2010-02-14 03:33:28.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2013-06-23 11:48:34.597709874 +0900
+@@ -54,7 +54,7 @@
+              'children' : children }
+ 
+ def main():
+-    from clang.cindex import Index
++    from clang30.cindex import Index
+     from pprint import pprint
+ 
+     from optparse import OptionParser, OptionGroup
+diff -urN clang-3.0.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py clang-3.0/tools/clang/bindings/python/examples/cindex/cindex-includes.py
+--- clang-3.0.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2010-02-14 03:33:28.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2013-06-23 11:48:47.417773464 +0900
+@@ -16,7 +16,7 @@
+ 
+ def main():
+     import sys
+-    from clang.cindex import Index
++    from clang30.cindex import Index
+ 
+     from optparse import OptionParser, OptionGroup
+ 
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_cursor.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py	2011-08-30 09:15:34.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_cursor.py	2013-06-23 11:49:44.790057960 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import Index, CursorKind, TypeKind
++from clang30.cindex import Index, CursorKind, TypeKind
+ 
+ kInput = """\
+ // FIXME: Find nicer way to drop builtins and other cruft.
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2011-07-06 12:00:34.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2013-06-23 11:49:05.649863858 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import CursorKind
++from clang30.cindex import CursorKind
+ 
+ def test_name():
+     assert CursorKind.UNEXPOSED_DECL.name is 'UNEXPOSED_DECL'
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_diagnostics.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2011-08-30 09:16:30.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2013-06-23 11:49:35.806013398 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang30.cindex import *
+ 
+ def tu_from_source(source):
+     index = Index.create()
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_index.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_index.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_index.py	2010-01-24 11:02:07.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_index.py	2013-06-23 11:49:25.717963371 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang30.cindex import *
+ import os
+ 
+ kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_location.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_location.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_location.py	2011-08-18 02:20:24.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_location.py	2013-06-23 11:49:17.037920328 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import Index
++from clang30.cindex import Index
+ 
+ baseInput="int one;\nint two;\n"
+ 
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_translation_unit.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2011-07-06 12:00:34.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-06-23 11:50:00.462135667 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import *
++from clang30.cindex import *
+ import os
+ 
+ kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
+diff -urN clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_type.py clang-3.0/tools/clang/bindings/python/tests/cindex/test_type.py
+--- clang-3.0.orig/tools/clang/bindings/python/tests/cindex/test_type.py	2011-10-22 20:36:45.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/tests/cindex/test_type.py	2013-06-23 11:49:53.510101187 +0900
+@@ -1,4 +1,4 @@
+-from clang.cindex import Index, CursorKind, TypeKind
++from clang30.cindex import Index, CursorKind, TypeKind
+ 
+ kInput = """\
+ 
diff -urN ../../clang-3.0/debian/patches/python-clangpath.diff clang-3.0/debian/patches/python-clangpath.diff
--- ../../clang-3.0/debian/patches/python-clangpath.diff	1970-01-01 09:00:00.000000000 +0900
+++ clang-3.0/debian/patches/python-clangpath.diff	2014-05-09 17:45:31.456192953 +0900
@@ -0,0 +1,12 @@
+diff -urN clang-3.0.orig/tools/clang/bindings/python/clang/cindex.py clang-3.0/tools/clang/bindings/python/clang/cindex.py
+--- clang-3.0.orig/tools/clang/bindings/python/clang/cindex.py	2011-10-22 20:36:45.000000000 +0900
++++ clang-3.0/tools/clang/bindings/python/clang/cindex.py	2013-06-23 11:48:10.085588327 +0900
+@@ -75,7 +75,7 @@
+     elif name == 'Windows':
+         return cdll.LoadLibrary('libclang.dll')
+     else:
+-        return cdll.LoadLibrary('libclang.so')
++        return cdll.LoadLibrary('libclang-3.0.so.1')
+ 
+ # ctypes doesn't implicitly convert c_void_p to the appropriate wrapper
+ # object. This is a problem, because it means that from_parameter will see an
diff -urN ../../clang-3.0/debian/patches/series clang-3.0/debian/patches/series
--- ../../clang-3.0/debian/patches/series	2013-03-28 18:01:26.000000000 +0900
+++ clang-3.0/debian/patches/series	2014-09-20 23:43:49.226880305 +0900
@@ -11,3 +11,6 @@
 28-gcc-4.7.diff
 29-set-default-cpu-for-armhf.diff
 30-powerpc-no-altivec.patch
+32-scan-build-path.diff
+python-clangpath.diff
+99-python-bindings.patch
diff -urN ../../clang-3.0/debian/rules clang-3.0/debian/rules
--- ../../clang-3.0/debian/rules	2013-02-06 19:13:50.000000000 +0900
+++ clang-3.0/debian/rules	2014-09-20 23:40:43.674782094 +0900
@@ -17,7 +17,7 @@
 
 define pkgname
   $(shell cat $(D)/debian/control.in/$1 | awk '/^Package: / {print $$2}' | \
-  sed -e 's/@UVERSION@/$(UVERSION)/;s/@GCC_VERSION@/$(gcc_version)/;s/@PKG_VERSION@/$(pkg_version)/')
+  sed -e 's/@UVERSION@/$(UVERSION)/;s/@GCC_VERSION@/$(GCC_VERSION)/;s/@PKG_VERSION@/$(pkg_version)/')
 endef
 
 define buildof
@@ -37,5 +37,7 @@
 	$(RM) -r $(D)/debian/ccache
 	find utils -name '*.py[co]' | xargs rm -f
 	$(RM) tools/clang/include/clang/Debian/debian_path.h
+	rm -rf tools/clang/tools/scan-build-$(UVERSION)
+	rm -rf tools/clang/tools/scan-view-$(UVERSION)
 
 .PHONY: clean
diff -urN ../../clang-3.0/debian/rules.d/binary.mk clang-3.0/debian/rules.d/binary.mk
--- ../../clang-3.0/debian/rules.d/binary.mk	2011-10-04 21:58:06.000000000 +0900
+++ clang-3.0/debian/rules.d/binary.mk	2014-09-20 23:41:26.653495347 +0900
@@ -17,6 +17,7 @@
 		-e "s/@PKG_VERSION@/$(pkg_version)/g" \
 		-e "s/@SHLIB_NAME@/$(shlib_name)/g" \
 		-e "s;@OCAML_STDLIB_DIR@;$(OCAML_STDLIB_DIR);g" \
+		-e "s;@DEB_HOST_MULTIARCH@;$(DEB_HOST_MULTIARCH);g" \
 		$(D)/debian/debhelper.in/$f > \
 		$(D)/debian/$(strip $(call pkgname,$*))$(strip $(suffix $(f))) \
 	&&) :
@@ -39,6 +40,10 @@
 	@echo Building package: $(call pkgname,$*)
 	dh_testdir
 	dh_testroot
+	if test $* = clang ; then \
+		cp -fR $(D)/tools/clang/tools/scan-build tools/clang/tools/scan-build-$(UVERSION) ; \
+		cp -fR $(D)/tools/clang/tools/scan-view tools/clang/tools/scan-view-$(UVERSION) ; \
+	fi
 	dh_installchangelogs -p$(call pkgname,$*)
 	dh_installdocs -p$(call pkgname,$*)
 	dh_installexamples -p$(call pkgname,$*)
@@ -73,6 +78,10 @@
 	dh_fixperms -p$(call pkgname,$*)
 	DH_VERBOSE=1 dh_makeshlibs -p$(call pkgname,$*)
 	DH_VERBOSE=1 dh_shlibdeps -p$(call pkgname,$*)
+	if test $* = python-clang ; then \
+		find $(D)/debian/python-clang-$(UVERSION)/usr/share/doc/python-clang-$(UVERSION)/examples -name "*.gz" -exec gzip -d {} \; ; \
+		find $(D)/debian/python-clang-$(UVERSION)/usr/share/doc/python-clang-$(UVERSION)/tests    -name "*.gz" -exec gzip -d {} \; ; \
+	fi
 	dh_installdeb -p$(call pkgname,$*)
 	dh_gencontrol -p$(call pkgname,$*)
 	dh_md5sums -p$(call pkgname,$*)
diff -urN ../../clang-3.0/debian/rules.d/build.mk clang-3.0/debian/rules.d/build.mk
--- ../../clang-3.0/debian/rules.d/build.mk	2012-02-25 22:05:16.000000000 +0900
+++ clang-3.0/debian/rules.d/build.mk	2014-05-09 17:45:31.456192953 +0900
@@ -74,10 +74,11 @@
 	    L=$${L##$(D)/debian/tmp-$*} ; \
 	    ln -sf $$L $$i ; \
 	  done
-	(cd $(D)/debian/tmp-clang/usr/lib/; \
-		if test ! -h libclang.so; then \
-			ln -s libclang.so.$(SONAME_EXT) libclang.so; \
-	fi)
+	cd $(D)/debian/tmp-clang/usr/lib/ && \
+	if test ! -f libclang-$(UVERSION).so.$(SONAME_EXT); then \
+		mv libclang.so.$(SONAME_EXT) libclang-$(UVERSION).so.$(SONAME_EXT) && \
+		ln -s libclang-$(UVERSION).so.$(SONAME_EXT) libclang.so.$(SONAME_EXT); \
+	fi
 	chrpath -d $(D)/debian/tmp-clang/usr/bin/clang
 	$(call $*_extra_install)
 	touch $@
diff -urN ../../clang-3.0/debian/rules.d/control.mk clang-3.0/debian/rules.d/control.mk
--- ../../clang-3.0/debian/rules.d/control.mk	2011-12-20 22:40:50.000000000 +0900
+++ clang-3.0/debian/rules.d/control.mk	2014-05-09 17:45:31.456192953 +0900
@@ -1,7 +1,7 @@
 $(foreach var,BUILDDEPS BUILDCONFLICTS,$(eval \
   $(if $(strip $($(var))),$(eval $(var) :=, $($(var))),$(eval $(var) :=))))
 
-extra_packages:=libclang1 libclang-dev libclang-common-dev
+extra_packages:=libclang1 libclang-dev libclang-common-dev python-clang
 packages := $(foreach build,$(builds),$($(build)_packages))
 packages += $(extra_packages) 
 
@@ -11,7 +11,7 @@
 	    -e "s/ *@BUILDDEPS@/$(BUILDDEPS)/g"			\
 	    -e "s/ *@BUILDCONFLICTS@/$(BUILDCONFLICTS)/g"	\
 	    -e "s/@UVERSION@/$(UVERSION)/g"			\
-	    -e "s/@GCC_VERSION@/$(gcc_version)/g"		\
+	    -e "s/@GCC_VERSION@/$(GCC_VERSION)/g"		\
 	    -e "s/@PKG_VERSION@/$(pkg_version)/g"		\
 	    -e "s/ , /, /g"					\
 	    $@.in/source $(addprefix $@.in/,$(packages)) > $@
diff -urN ../../clang-3.0/debian/rules.d/vars.mk clang-3.0/debian/rules.d/vars.mk
--- ../../clang-3.0/debian/rules.d/vars.mk	2011-12-08 02:05:16.000000000 +0900
+++ clang-3.0/debian/rules.d/vars.mk	2014-05-09 17:45:31.456192953 +0900
@@ -5,6 +5,7 @@
 PATCHSETVERSION	:= $(shell echo $(VERSION) | cut -d'-' -f2)
 UVERSION	:= 3.0
 SONAME_EXT      := 1
+GCC_VERSION	:= 4.6
 OS		:= $(shell lsb_release -is)
 PF		:= /usr/lib/llvm-$(UVERSION)
 D		:= $(CURDIR)
@@ -18,7 +19,7 @@
 SUFFIX		:= $(subst $(PKGNAME),,$(PKGSOURCE))
 
 # for the shared lib
-pkg_version	:= $(UVERSION)-3
+pkg_version	:= $(UVERSION)
 shlib_name	:= libLLVM-$(pkg_version)rc
 
 # for llvm-gcc and tools
@@ -59,10 +60,10 @@
 endif
 
 confargs := \
-	CC=$(DEB_HOST_GNU_TYPE)-gcc CXX=$(DEB_HOST_GNU_TYPE)-g++ \
-	CPP=$(DEB_HOST_GNU_TYPE)-cpp \
+	CC=$(DEB_HOST_GNU_TYPE)-gcc-$(GCC_VERSION) CXX=$(DEB_HOST_GNU_TYPE)-g++-$(GCC_VERSION) \
+	CPP=$(DEB_HOST_GNU_TYPE)-cpp-$(GCC_VERSION) \
 	--with-c-include-dirs=/usr/include/$(DEB_HOST_GNU_TYPE):/usr/include \
-    --with-cxx-include-root=/usr/include/c++/4.6 \
+    --with-cxx-include-root=/usr/include/c++/$(GCC_VERSION) \
     --with-cxx-include-arch=$(DEB_HOST_GNU_TYPE) \
     --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE)
 
