diff -urN clang-3.1.orig/tools/clang/bindings/python/README.txt clang-3.1/tools/clang/bindings/python/README.txt
--- clang-3.1.orig/tools/clang/bindings/python/README.txt	2010-05-01 06:51:10.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/README.txt	2013-06-23 12:37:19.028211350 +0900
@@ -4,12 +4,9 @@
 
 This directory implements Python bindings for Clang.
 
-You may need to alter LD_LIBRARY_PATH so that the Clang library can be
-found. The unit tests are designed to be run with 'nosetests'. For example:
+The unit tests are designed to be run with 'nosetests'. For example:
 --
-$ env PYTHONPATH=$(echo ~/llvm/tools/clang/bindings/python/) \
-      LD_LIBRARY_PATH=$(llvm-config --libdir) \
-  nosetests -v
+$ nosetests -v
 tests.cindex.test_index.test_create ... ok
 ...
 
diff -urN clang-3.1.orig/tools/clang/bindings/python/clang/cindex.py clang-3.1/tools/clang/bindings/python/clang/cindex.py
--- clang-3.1.orig/tools/clang/bindings/python/clang/cindex.py	2012-04-16 03:51:10.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/clang/cindex.py	2013-06-23 12:41:27.673444311 +0900
@@ -76,7 +76,7 @@
     elif name == 'Windows':
         return cdll.LoadLibrary('libclang.dll')
     else:
-        return cdll.LoadLibrary('libclang.so')
+        return cdll.LoadLibrary('/usr/lib/llvm-3.1/lib/libclang.so.1')
 
 # ctypes doesn't implicitly convert c_void_p to the appropriate wrapper
 # object. This is a problem, because it means that from_parameter will see an
diff -urN clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-dump.py
--- clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2010-02-14 03:33:28.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2013-06-23 12:37:48.156355790 +0900
@@ -54,7 +54,7 @@
              'children' : children }
 
 def main():
-    from clang.cindex import Index
+    from clang31.cindex import Index
     from pprint import pprint
 
     from optparse import OptionParser, OptionGroup
diff -urN clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-includes.py
--- clang-3.1.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2010-02-14 03:33:28.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2013-06-23 12:37:59.980414432 +0900
@@ -16,7 +16,7 @@
 
 def main():
     import sys
-    from clang.cindex import Index
+    from clang31.cindex import Index
 
     from optparse import OptionParser, OptionGroup
 
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py	2012-03-11 07:23:27.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor.py	2013-06-23 12:39:15.172787294 +0900
@@ -1,5 +1,5 @@
-from clang.cindex import CursorKind
-from clang.cindex import TypeKind
+from clang31.cindex import CursorKind
+from clang31.cindex import TypeKind
 from .util import get_cursor
 from .util import get_tu
 
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2012-02-05 20:42:09.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2013-06-23 12:38:14.772487787 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import CursorKind
+from clang31.cindex import CursorKind
 
 def test_name():
     assert CursorKind.UNEXPOSED_DECL.name is 'UNEXPOSED_DECL'
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_diagnostics.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2012-03-11 07:19:05.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2013-06-23 12:39:03.356728676 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang31.cindex import *
 from .util import get_tu
 
 # FIXME: We need support for invalid translation units to test better.
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_file.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_file.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_file.py	2011-10-31 09:07:19.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_file.py	2013-06-23 12:39:27.436848105 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import Index, File
+from clang31.cindex import Index, File
 
 def test_file():
   index = Index.create()
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_index.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_index.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_index.py	2010-01-24 11:02:07.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_index.py	2013-06-23 12:38:51.524670030 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang31.cindex import *
 import os
 
 kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_location.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_location.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_location.py	2012-03-11 07:19:05.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_location.py	2013-06-23 12:38:28.228554488 +0900
@@ -1,7 +1,7 @@
-from clang.cindex import Cursor
-from clang.cindex import File
-from clang.cindex import SourceLocation
-from clang.cindex import SourceRange
+from clang31.cindex import Cursor
+from clang31.cindex import File
+from clang31.cindex import SourceLocation
+from clang31.cindex import SourceRange
 from .util import get_cursor
 from .util import get_tu
 
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_translation_unit.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2011-07-06 12:00:34.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-06-23 12:39:50.340961675 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang31.cindex import *
 import os
 
 kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_type.py clang-3.1/tools/clang/bindings/python/tests/cindex/test_type.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/test_type.py	2012-04-16 03:51:10.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/test_type.py	2013-06-23 12:39:38.620903555 +0900
@@ -1,5 +1,5 @@
-from clang.cindex import CursorKind
-from clang.cindex import TypeKind
+from clang31.cindex import CursorKind
+from clang31.cindex import TypeKind
 from nose.tools import raises
 from .util import get_cursor
 from .util import get_tu
diff -urN clang-3.1.orig/tools/clang/bindings/python/tests/cindex/util.py clang-3.1/tools/clang/bindings/python/tests/cindex/util.py
--- clang-3.1.orig/tools/clang/bindings/python/tests/cindex/util.py	2012-03-11 07:19:05.000000000 +0900
+++ clang-3.1/tools/clang/bindings/python/tests/cindex/util.py	2013-06-23 12:38:41.956622576 +0900
@@ -1,7 +1,7 @@
 # This file provides common utility functions for the test suite.
 
-from clang.cindex import Cursor
-from clang.cindex import Index
+from clang31.cindex import Cursor
+from clang31.cindex import Index
 
 def get_tu(source, lang='c', all_warnings=False):
     """Obtain a translation unit from source and language.
