Index: llvm-toolchain_3.3~svn179457/autoconf/configure.ac
===================================================================
--- llvm-toolchain_3.3~svn179457.orig/autoconf/configure.ac	2013-03-25 14:13:33.000000000 +0100
+++ llvm-toolchain_3.3~svn179457/autoconf/configure.ac	2013-04-13 10:29:43.485931880 +0200
@@ -31,7 +31,7 @@
 dnl===-----------------------------------------------------------------------===
 dnl Initialize autoconf and define the package name, version number and
 dnl address for reporting bugs.
-AC_INIT([LLVM],[3.3svn],[http://llvm.org/bugs/])
+AC_INIT([LLVM],[3.3],[http://llvm.org/bugs/])
 AC_DEFINE([LLVM_VERSION_MAJOR], [3], [Major version of the LLVM API])
 AC_DEFINE([LLVM_VERSION_MINOR], [3], [Minor version of the LLVM API])
 
Index: llvm-toolchain_3.3~svn179457/configure
===================================================================
--- llvm-toolchain_3.3~svn179457.orig/configure	2013-04-13 10:29:39.953932017 +0200
+++ llvm-toolchain_3.3~svn179457/configure	2013-04-13 10:29:43.493931880 +0200
@@ -561,8 +561,8 @@
 # Identity of this package.
 PACKAGE_NAME='LLVM'
 PACKAGE_TARNAME='llvm'
-PACKAGE_VERSION='3.3svn'
-PACKAGE_STRING='LLVM 3.3svn'
+PACKAGE_VERSION='3.3'
+PACKAGE_STRING='LLVM 3.3'
 PACKAGE_BUGREPORT='http://llvm.org/bugs/'
 
 ac_unique_file="lib/IR/Module.cpp"
