Index: llvm-toolchain-3.3-3.3/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp
===================================================================
--- llvm-toolchain-3.3-3.3.orig/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp	2013-06-20 10:10:31.000000000 +0200
+++ llvm-toolchain-3.3-3.3/lldb/source/Plugins/Process/Linux/LinuxSignals.cpp	2013-06-20 10:11:57.000000000 +0200
@@ -45,7 +45,9 @@
     ADDSIGNAL(PIPE,   false,  true,  true, "write to pipe with reading end closed");
     ADDSIGNAL(ALRM,   false,  false, true, "alarm");
     ADDSIGNAL(TERM,   false,  true,  true, "termination requested");
+#ifdef SIGSTKFLT
     ADDSIGNAL(STKFLT, false,  true,  true, "stack fault");
+#endif
     ADDSIGNAL(CHLD,   false,  false, true, "child process exit");
     ADDSIGNAL(CONT,   false,  true,  true, "process continue");
     ADDSIGNAL(STOP,   false,  true,  true, "process stop");
Index: llvm-toolchain-3.3-3.3/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp
===================================================================
--- llvm-toolchain-3.3-3.3.orig/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp	2013-06-20 10:10:31.000000000 +0200
+++ llvm-toolchain-3.3-3.3/lldb/source/Plugins/Process/Utility/RegisterContextMacOSXFrameBackchain.cpp	2013-06-20 10:11:57.000000000 +0200
@@ -149,7 +149,7 @@
 
             // TOOD: need a better way to detect when "long double" types are 
             // the same bytes size as "double"
-#if !defined(__arm__)
+#if !defined(__arm__) && !defined(__mips__)
         case sizeof (long double):
             if (sizeof (long double) == sizeof(uint32_t))
             {
