Index: llvm-toolchain-snapshot_3.3~svn181457/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.3~svn181457.orig/clang/tools/scan-build/scan-build	2013-05-02 02:52:46.000000000 +0200
+++ llvm-toolchain-snapshot_3.3~svn181457/clang/tools/scan-build/scan-build	2013-05-08 20:27:18.000000000 +0200
@@ -1524,7 +1524,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
