diff -urN clang-2.9.orig/tools/clang/bindings/python/README.txt clang-2.9/tools/clang/bindings/python/README.txt
--- clang-2.9.orig/tools/clang/bindings/python/README.txt	2010-05-01 06:51:10.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/README.txt	2013-06-23 03:01:45.287654760 +0900
@@ -4,12 +4,9 @@
 
 This directory implements Python bindings for Clang.
 
-You may need to alter LD_LIBRARY_PATH so that the Clang library can be
-found. The unit tests are designed to be run with 'nosetests'. For example:
+The unit tests are designed to be run with 'nosetests'. For example:
 --
-$ env PYTHONPATH=$(echo ~/llvm/tools/clang/bindings/python/) \
-      LD_LIBRARY_PATH=$(llvm-config --libdir) \
-  nosetests -v
+$ nosetests -v
 tests.cindex.test_index.test_create ... ok
 ...
 
diff -urN clang-2.9.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py clang-2.9/tools/clang/bindings/python/examples/cindex/cindex-dump.py
--- clang-2.9.orig/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2010-02-14 03:33:28.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/examples/cindex/cindex-dump.py	2013-06-23 02:11:19.975599560 +0900
@@ -54,7 +54,7 @@
              'children' : children }
 
 def main():
-    from clang.cindex import Index
+    from clang29.cindex import Index
     from pprint import pprint
 
     from optparse import OptionParser, OptionGroup
diff -urN clang-2.9.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py clang-2.9/tools/clang/bindings/python/examples/cindex/cindex-includes.py
--- clang-2.9.orig/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2010-02-14 03:33:28.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/examples/cindex/cindex-includes.py	2013-06-23 02:11:33.111664694 +0900
@@ -16,7 +16,7 @@
 
 def main():
     import sys
-    from clang.cindex import Index
+    from clang29.cindex import Index
 
     from optparse import OptionParser, OptionGroup
 
diff -urN clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py clang-2.9/tools/clang/bindings/python/tests/cindex/test_cursor.py
--- clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_cursor.py	2010-01-31 08:58:39.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/tests/cindex/test_cursor.py	2013-06-23 02:13:45.028318843 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import Index, CursorKind
+from clang29.cindex import Index, CursorKind
 
 kInput = """\
 // FIXME: Find nicer way to drop builtins and other cruft.
diff -urN clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py clang-2.9/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py
--- clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2010-01-25 06:20:39.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/tests/cindex/test_cursor_kind.py	2013-06-23 02:13:19.692193191 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import CursorKind
+from clang29.cindex import CursorKind
 
 def test_name():
     assert CursorKind.UNEXPOSED_DECL.name is 'UNEXPOSED_DECL'
diff -urN clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py clang-2.9/tools/clang/bindings/python/tests/cindex/test_diagnostics.py
--- clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2011-02-06 02:53:53.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/tests/cindex/test_diagnostics.py	2013-06-23 02:13:36.044274289 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang29.cindex import *
 
 def tu_from_source(source):
     index = Index.create()
diff -urN clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_index.py clang-2.9/tools/clang/bindings/python/tests/cindex/test_index.py
--- clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_index.py	2010-01-24 11:02:07.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/tests/cindex/test_index.py	2013-06-23 02:13:28.396236334 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang29.cindex import *
 import os
 
 kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
diff -urN clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py clang-2.9/tools/clang/bindings/python/tests/cindex/test_translation_unit.py
--- clang-2.9.orig/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2011-02-06 02:54:00.000000000 +0900
+++ clang-2.9/tools/clang/bindings/python/tests/cindex/test_translation_unit.py	2013-06-23 02:13:52.964358189 +0900
@@ -1,4 +1,4 @@
-from clang.cindex import *
+from clang29.cindex import *
 import os
 
 kInputsDir = os.path.join(os.path.dirname(__file__), 'INPUTS')
