diff -urN debian/changelog llvm-3.2.src/debian/changelog
--- debian/changelog	2012-12-24 06:51:37.000000000 +0900
+++ llvm-3.2.src/debian/changelog	2013-01-11 13:19:57.549488554 +0900
@@ -1,3 +1,9 @@
+llvm-3.2 (3.2-2ppa1) precise; urgency=low
+
+  * ppa for precise
+
+ -- ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>  Fri, 11 Jan 2013 13:57:50 +0900
+
 llvm-3.2 (3.2-2) unstable; urgency=low
 
   * Bring back manpages + html documentation (now based on sphinx)
diff -urN debian/control llvm-3.2.src/debian/control
--- debian/control	2012-12-24 15:06:45.000000000 +0900
+++ llvm-3.2.src/debian/control	2013-01-12 18:28:22.735400434 +0900
@@ -1,8 +1,7 @@
 Source: llvm-3.2
 Section: devel
-Priority: optional
-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
-Uploaders: Sylvestre Ledru <sylvestre@debian.org>
+Priority: extra
+Maintainer: ABE Hiroki (hATrayflood) <h.rayflood@gmail.com>
 Build-Depends: debhelper (>= 9), flex, bison, dejagnu, tcl8.5, expect,
     autoconf, automake1.9, perl, libtool, doxygen, chrpath, texinfo, quilt,
     sharutils, autotools-dev (>= 20060702.1), libffi-dev (>= 3.0.9),
@@ -10,9 +9,8 @@
     binutils-dev (>= 2.20.51.20100222-1~), ocaml-nox (>= 3.11.2), ocaml-best-compilers | ocaml-nox, dh-ocaml (>= 0.9.1)
 Build-Conflicts: oprofile
 Standards-Version: 3.9.3
-Homepage: http://www.llvm.org/
-Vcs-Svn: svn://svn.debian.org/svn/pkg-llvm/llvm/trunk/
-Vcs-Browser: http://svn.debian.org/viewsvn/pkg-llvm/llvm/trunk/
+Homepage: https://launchpad.net/~h-rayflood/+archive/llvm
+Vcs-Browser: https://github.com/hATrayflood/llvm-ppa
 
 
 Package: libllvm3.2
@@ -28,20 +26,6 @@
  .
  This package contains the LLVM runtime library.
 
-Package: libllvm3.2-dbg
-Architecture: any
-Section: debug
-Priority: extra
-Depends: ${shlibs:Depends}, ${misc:Depends}, libllvm3.2
-Pre-Depends: ${misc:Pre-Depends}
-Multi-Arch: same
-Description: Low-Level Virtual Machine (LLVM), debugging symbols library
- The Low-Level Virtual Machine (LLVM) is a collection of libraries and
- tools that make it easy to build compilers, optimizers, Just-In-Time
- code generators, and many other compiler-related programs.
- .
- This package contains the LLVM runtime library debug symbols.
-
 Package: llvm-3.2
 Architecture: any
 Suggests: llvm-3.2-doc
diff -urN debian/libllvm-3.2-ocaml-dev.dirs.in llvm-3.2.src/debian/libllvm-3.2-ocaml-dev.dirs.in
--- debian/libllvm-3.2-ocaml-dev.dirs.in	2012-12-24 05:55:05.000000000 +0900
+++ llvm-3.2.src/debian/libllvm-3.2-ocaml-dev.dirs.in	2013-01-12 18:17:25.128139529 +0900
@@ -1,2 +1,2 @@
-@OCAML_STDLIB_DIR@/META/
+@OCAML_STDLIB_DIR@/METAS
 
diff -urN debian/rules llvm-3.2.src/debian/rules
--- debian/rules	2012-12-24 05:55:05.000000000 +0900
+++ llvm-3.2.src/debian/rules	2013-01-12 22:12:34.897376653 +0900
@@ -28,7 +28,7 @@
 debian/%: debian/%.in
 	sed -e 's|@DEB_HOST_MULTIARCH@|$(DEB_HOST_MULTIARCH)|g' -e "s|@OCAML_STDLIB_DIR@|$(OCAML_STDLIB_DIR)|g" -e "s|@LLVM_VERSION@|$(LLVM_VERSION)|g" $< > $@
 
-override_dh_auto_configure: debian/libllvm$(LLVM_VERSION).install debian/llvm-$(LLVM_VERSION)-dev.links debian/libllvm-$(LLVM_VERSION)-ocaml-dev.dirs debian/llvm-$(LLVM_VERSION)-runtime.binfmt debian/llvm-$(LLVM_VERSION)-runtime.prerm
+override_dh_auto_configure: debian/libllvm$(LLVM_VERSION).install debian/llvm-$(LLVM_VERSION)-dev.links debian/libllvm-$(LLVM_VERSION)-ocaml-dev.dirs debian/llvm-$(LLVM_VERSION)-runtime.binfmt debian/llvm-$(LLVM_VERSION)-runtime.prerm debian/libllvm-$(LLVM_VERSION)-ocaml-dev.install
 	mkdir -p $(TARGET_BUILD)
 	echo "#define LLVM_DEBIAN_INFO \"$(DISTRELEASE) $(LLVM_FULL_VERSION)\"" > $(TARGET_BUILD)/debian.h
 	cd $(TARGET_BUILD) && \
@@ -84,14 +84,14 @@
 	for f in *.1; do \
 		echo "$$f"|grep "3\.2" || mv $$f `echo $$f|sed "s|\.1|-$(LLVM_VERSION).1|"`; \
 	done
+
+	cp $(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev.META \
+           $(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev/$(OCAML_STDLIB_DIR)/METAS/META.llvm-$(subst .,_,$(LLVM_VERSION))
 	dh_auto_install
 
 override_dh_shlibdeps:
 	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(CURDIR)/debian/tmp/usr/lib/llvm-$(LLVM_VERSION)/lib/ dh_shlibdeps
 
-override_dh_strip:
-	dh_strip --dbg-package=libllvm$(LLVM_VERSION)-dbg
-
 override_dh_installdeb:
 	# Belong to llvm runtime
 	rm -f $(CURDIR)/debian/llvm-$(LLVM_VERSION)/usr/share/man/man1/lli*
@@ -109,3 +109,10 @@
 	rm -rf $(TARGET_BUILD)
 	find utils -name '*.pyc' | xargs -r rm -f
 	find test -name '*.pyc' -o -name '*.o' -o -name '*.cm[ix]' | xargs -r rm -f
+	rm -rf docs/_build
+	rm -f $(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev.dirs
+	rm -f $(CURDIR)/debian/libllvm-$(LLVM_VERSION)-ocaml-dev.install
+	rm -f $(CURDIR)/debian/libllvm$(LLVM_VERSION).install
+	rm -f $(CURDIR)/debian/llvm-$(LLVM_VERSION)-dev.links
+	rm -f $(CURDIR)/debian/llvm-$(LLVM_VERSION)-runtime.binfmt
+	rm -f $(CURDIR)/debian/llvm-$(LLVM_VERSION)-runtime.prerm
