Index: llvm-toolchain-3.4-3.4.2/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-3.4-3.4.2.orig/clang/lib/Driver/ToolChains.cpp
+++ llvm-toolchain-3.4-3.4.2/clang/lib/Driver/ToolChains.cpp
@@ -1124,14 +1124,15 @@ void Generic_GCC::GCCInstallationDetecto
   static const char *const X86_64Triples[] = {
     "x86_64-linux-gnu", "x86_64-unknown-linux-gnu", "x86_64-pc-linux-gnu",
     "x86_64-redhat-linux6E", "x86_64-redhat-linux", "x86_64-suse-linux",
-    "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux"
+    "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux",
+    "x86_64-kfreebsd-gnu"
   };
   static const char *const X86LibDirs[] = { "/lib32", "/lib" };
   static const char *const X86Triples[] = {
     "i686-linux-gnu", "i686-pc-linux-gnu", "i486-linux-gnu", "i386-linux-gnu",
     "i386-redhat-linux6E", "i686-redhat-linux", "i586-redhat-linux",
     "i386-redhat-linux", "i586-suse-linux", "i486-slackware-linux",
-    "i686-montavista-linux"
+    "i686-montavista-linux", "i486-kfreebsd-gnu", "i586-kfreebsd-gnu"
   };
 
   static const char *const MIPSLibDirs[] = { "/lib" };
@@ -2588,6 +2589,10 @@ void Linux::AddClangSystemIncludeArgs(co
   if (DriverArgs.hasArg(options::OPT_nostdinc))
     return;
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+
   if (!DriverArgs.hasArg(options::OPT_nostdlibinc))
     addSystemInclude(DriverArgs, CC1Args, SysRoot + "/usr/local/include");
 
Index: llvm-toolchain-3.4-3.4.2/lldb/source/Host/common/Host.cpp
===================================================================
--- llvm-toolchain-3.4-3.4.2.orig/lldb/source/Host/common/Host.cpp
+++ llvm-toolchain-3.4-3.4.2/lldb/source/Host/common/Host.cpp
@@ -1586,7 +1586,7 @@ Host::RunShellCommand (const char *comma
     return error;
 }
 
-#if defined(__linux__) or defined(__FreeBSD__)
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__GLIBC__)
 // The functions below implement process launching via posix_spawn() for Linux
 // and FreeBSD.
 
