Index: llvm-toolchain-3.4-3.4.2/utils/TableGen/InstrInfoEmitter.cpp
===================================================================
--- llvm-toolchain-3.4-3.4.2.orig/utils/TableGen/InstrInfoEmitter.cpp
+++ llvm-toolchain-3.4-3.4.2/utils/TableGen/InstrInfoEmitter.cpp
@@ -569,6 +569,9 @@ void InstrInfoEmitter::emitEnums(raw_ost
   OS << "\n#ifdef GET_INSTRINFO_ENUM\n";
   OS << "#undef GET_INSTRINFO_ENUM\n";
 
+  OS << "// Undef for HURD\n";
+  OS << "#ifdef EIEIO\n#undef EIEIO\n#endif\n";
+
   OS << "namespace llvm {\n\n";
 
   CodeGenTarget Target(Records);
