Index: llvm-toolchain-3.4-3.4.2/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-3.4-3.4.2.orig/clang/tools/scan-build/scan-build
+++ llvm-toolchain-3.4-3.4.2/clang/tools/scan-build/scan-build
@@ -1547,7 +1547,7 @@ if (!@ARGV and !$RequestDisplayHelp) {
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/lib/llvm-3.4/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
