Index: llvm-toolchain-snapshot_3.4~svn193982/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193982.orig/tools/llvm-shlib/Makefile	2013-11-04 09:46:03.312667676 +0100
+++ llvm-toolchain-snapshot_3.4~svn193982/tools/llvm-shlib/Makefile	2013-11-04 09:46:03.308667675 +0100
@@ -76,7 +76,7 @@
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), DragonFly Linux FreeBSD GNU/kFreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU GNU/kFreeBSD))
