Index: llvm-toolchain-3.4-3.4~+rc1/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/clang/lib/Driver/ToolChains.cpp	2013-12-02 10:15:22.000000000 +0100
+++ llvm-toolchain-3.4-3.4~+rc1/clang/lib/Driver/ToolChains.cpp	2013-12-02 10:16:54.648134997 +0100
@@ -2618,6 +2618,11 @@
     return;
   }
 
+  // Force the inclusion of the gcc headers (objc/objc.h)
+  addExternCSystemIncludeIfExists(
+     DriverArgs, CC1Args, GCCInstallation.getInstallPath() + "/include");
+
+
   // Lacking those, try to detect the correct set of system includes for the
   // target triple.
 
