Index: llvm-toolchain-3.4-3.4.1/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-3.4-3.4.1.orig/clang/lib/Driver/ToolChains.cpp	2014-04-26 01:38:13.000000000 +0200
+++ llvm-toolchain-3.4-3.4.1/clang/lib/Driver/ToolChains.cpp	2014-05-10 12:34:16.374471699 +0200
@@ -2588,6 +2588,10 @@
   if (DriverArgs.hasArg(options::OPT_nostdinc))
     return;
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+
   if (!DriverArgs.hasArg(options::OPT_nostdlibinc))
     addSystemInclude(DriverArgs, CC1Args, SysRoot + "/usr/local/include");
 
Index: llvm-toolchain-3.4-3.4.1/lldb/source/Host/common/Host.cpp
===================================================================
--- llvm-toolchain-3.4-3.4.1.orig/lldb/source/Host/common/Host.cpp	2013-12-16 10:49:51.000000000 +0100
+++ llvm-toolchain-3.4-3.4.1/lldb/source/Host/common/Host.cpp	2014-05-10 12:33:20.000000000 +0200
@@ -1586,7 +1586,7 @@
     return error;
 }
 
-#if defined(__linux__) or defined(__FreeBSD__)
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__GLIBC__)
 // The functions below implement process launching via posix_spawn() for Linux
 // and FreeBSD.
 
