Index: llvm-toolchain-snapshot_3.5~svn201651/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.5~svn201651.orig/clang/lib/Driver/ToolChains.cpp	2014-02-19 09:46:35.377059510 +0100
+++ llvm-toolchain-snapshot_3.5~svn201651/clang/lib/Driver/ToolChains.cpp	2014-02-19 09:46:39.989351664 +0100
@@ -3241,7 +3241,7 @@
 
 #ifdef __i386__
   // clang does not detect this path: /usr/include/i386-linux-gnu/c++/4.8/ Help him
-  addLibStdCXXIncludePaths( "/usr/include/", "/c++/" + Version.str(), "i386-linux-gnu", GCCInstallation.getMultiarchSuffix(), DriverArgs, CC1Args);
+  addLibStdCXXIncludePaths( "/usr/include/", "/c++/" + Version.Text, "i386-linux-gnu", Multilib.includeSuffix(), DriverArgs, CC1Args);
 #endif
 
   if (addLibStdCXXIncludePaths(LibDir.str() + "/../include",
