Index: llvm-toolchain-3.5-3.5~+rc2/lldb/tools/Makefile
===================================================================
--- llvm-toolchain-3.5-3.5~+rc2.orig/lldb/tools/Makefile
+++ llvm-toolchain-3.5-3.5~+rc2/lldb/tools/Makefile
@@ -13,7 +13,7 @@ include $(LLDB_LEVEL)/../../Makefile.con
 DIRS :=
 
 # enable lldb-gdbserver for supported platforms
-ifneq (,$(strip $(filter $(HOST_OS), FreeBSD Linux)))
+ifneq (,$(strip $(filter $(HOST_OS), FreeBSD Linux GNU/kFreeBSD)))
 DIRS += lldb-gdbserver
 endif
 
