Index: llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-snapshot_3.4~svn193982.orig/clang/lib/Driver/ToolChains.cpp	2013-11-04 09:47:04.468669903 +0100
+++ llvm-toolchain-snapshot_3.4~svn193982/clang/lib/Driver/ToolChains.cpp	2013-11-04 09:47:04.464669903 +0100
@@ -2529,6 +2529,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
