Index: llvm-toolchain-3.4-3.4~+rc1/clang/lib/Basic/FileManager.cpp
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/clang/lib/Basic/FileManager.cpp	2013-11-26 15:06:44.000000000 +0100
+++ llvm-toolchain-3.4-3.4~+rc1/clang/lib/Basic/FileManager.cpp	2013-11-27 15:37:41.000000000 +0100
@@ -43,6 +43,11 @@
 #if defined(LLVM_ON_UNIX)
 #include <limits.h>
 #endif
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 using namespace clang;
 
 // FIXME: Enhance libsystem to support inode and other fields.
Index: llvm-toolchain-3.4-3.4~+rc1/lldb/include/lldb/lldb-defines.h
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/lldb/include/lldb/lldb-defines.h	2013-11-26 15:06:44.000000000 +0100
+++ llvm-toolchain-3.4-3.4~+rc1/lldb/include/lldb/lldb-defines.h	2013-11-27 15:37:41.000000000 +0100
@@ -12,6 +12,11 @@
 
 #include "lldb/lldb-types.h"
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 #if !defined(UINT32_MAX)
     #define UINT32_MAX 4294967295U
 #endif
Index: llvm-toolchain-3.4-3.4~+rc1/lib/Support/Unix/Path.inc
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/lib/Support/Unix/Path.inc	2013-10-31 15:35:00.000000000 +0100
+++ llvm-toolchain-3.4-3.4~+rc1/lib/Support/Unix/Path.inc	2013-11-27 17:31:18.478743021 +0100
@@ -59,6 +59,7 @@
 // For GNU Hurd
 #if defined(__GNU__) && !defined(PATH_MAX)
 # define PATH_MAX 4096
+# define MAXPATHLEN 4096
 #endif
 
 using namespace llvm;
