Index: llvm-toolchain-snapshot_3.4~svn192346/clang/tools/scan-build/scan-build
===================================================================
--- llvm-toolchain-snapshot_3.4~svn192346.orig/clang/tools/scan-build/scan-build	2013-10-10 13:38:20.858082707 +0200
+++ llvm-toolchain-snapshot_3.4~svn192346/clang/tools/scan-build/scan-build	2013-10-10 13:38:20.854082707 +0200
@@ -1547,7 +1547,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
