Index: llvm-toolchain-3.4-3.4~+rc1/lldb/source/Interpreter/Makefile
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/lldb/source/Interpreter/Makefile	2013-08-23 14:44:05.000000000 +0200
+++ llvm-toolchain-3.4-3.4~+rc1/lldb/source/Interpreter/Makefile	2013-11-28 11:23:01.664429006 +0100
@@ -23,7 +23,7 @@
 
 # Drop -Wself-assign, -Wmissing-field-initializers and -Wsometimes-uninitialized,
 # which we are not currently clean with (due to SWIG generated cpp source).
-EXTRA_OPTIONS += -Wno-missing-field-initializers -Wno-self-assign -Wno-sometimes-uninitialized
+EXTRA_OPTIONS += -Wno-missing-field-initializers -Wno-self-assign -Wno-sometimes-uninitialized -Wno-cast-qual
 
 # edit-swig-python-wrapper-file.py needs $(SRCROOT)
 export SRCROOT := $(PROJ_SRC_DIR)/$(LLDB_LEVEL)
