Index: clang-3.1~+rc3/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.1~+rc3.orig/tools/clang/lib/Driver/ToolChains.cpp	2012-05-20 16:58:56.350226263 +0200
+++ clang-3.1~+rc3/tools/clang/lib/Driver/ToolChains.cpp	2012-05-20 16:59:06.134226139 +0200
@@ -2155,6 +2155,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
