Index: llvm-toolchain-3.4-3.4~+rc1/clang/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-toolchain-3.4-3.4~+rc1.orig/clang/lib/Driver/ToolChains.cpp	2013-11-30 15:03:35.257604848 +0000
+++ llvm-toolchain-3.4-3.4~+rc1/clang/lib/Driver/ToolChains.cpp	2013-11-30 15:04:14.793603425 +0000
@@ -2771,6 +2771,11 @@
   StringRef BiarchSuffix = GCCInstallation.getBiarchSuffix();
   const GCCVersion &Version = GCCInstallation.getVersion();
 
+#ifdef __i386__
+  // clang does not detect this path: /usr/include/i386-linux-gnu/c++/4.8/ Help him
+  addLibStdCXXIncludePaths( "/usr/include/", "/c++/" + Version.str(), "i386-linux-gnu", GCCInstallation.getMultiarchSuffix(), DriverArgs, CC1Args);
+#endif
+
   if (addLibStdCXXIncludePaths(LibDir.str() + "/../include",
                                "/c++/" + Version.Text, TripleStr, BiarchSuffix,
                                MIPSABIDirSuffix, DriverArgs, CC1Args))
