Index: clang-3.0/tools/clang/Makefile
===================================================================
--- clang-3.0.orig/tools/clang/Makefile	2011-12-05 01:00:56.742977627 +0100
+++ clang-3.0/tools/clang/Makefile	2011-12-05 01:01:07.266977393 +0100
@@ -43,9 +43,7 @@
 
 # Set common Clang build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(CLANG_LEVEL)/include -I$(PROJ_OBJ_DIR)/$(CLANG_LEVEL)/include
-ifdef CLANG_VENDOR
-CPP.Flags += -DCLANG_VENDOR='"$(CLANG_VENDOR) "'
-endif
+CPP.Flags += -DCLANG_VENDOR='"Debian "'
 ifdef CLANG_REPOSITORY_STRING
 CPP.Flags += -DCLANG_REPOSITORY_STRING='"$(CLANG_REPOSITORY_STRING)"'
 endif
