Index: llvm-toolchain-3.6-3.6/lldb/source/Core/Module.cpp
===================================================================
--- llvm-toolchain-3.6-3.6.orig/lldb/source/Core/Module.cpp
+++ llvm-toolchain-3.6-3.6/lldb/source/Core/Module.cpp
@@ -1606,8 +1606,8 @@ Module::SetArchitecture (const ArchSpec
     {
         m_arch = new_arch;
         return true;
-    }    
-    return m_arch.IsExactMatch(new_arch);
+    }
+    return m_arch.IsCompatibleMatch(new_arch);
 }
 
 bool 
