Description: Make sure PowerPC doesn't default to altivec on
Author: Adam Conrad <adconrad@ubuntu.com>
Author: Peter Michael Green <plugwash@debian.org>
Forwarded: no
Reviewed-By: Colin Watson <cjwatson@ubuntu.com>

Last-Update: 2013-04-01

Index: clang-3.2.new/clang/lib/Lex/Makefile
===================================================================
--- clang-3.2.new.orig/clang/lib/Lex/Makefile	2013-04-01 02:32:35.000000000 +0000
+++ clang-3.2.new/clang/lib/Lex/Makefile	2013-04-01 02:32:03.000000000 +0000
@@ -16,11 +16,5 @@
 
 LIBRARYNAME := clangLex
 
-ifeq ($(ARCH),PowerPC)
-ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),powerpcspe)
-CXX.Flags += -maltivec
-endif
-endif
-
 include $(CLANG_LEVEL)/Makefile
 
