Index: llvm-toolchain-3.2-3.2repack/lldb/source/Host/common/Host.cpp
===================================================================
--- llvm-toolchain-3.2-3.2repack.orig/lldb/source/Host/common/Host.cpp	2013-05-15 15:22:39.545049856 +0200
+++ llvm-toolchain-3.2-3.2repack/lldb/source/Host/common/Host.cpp	2013-05-15 15:23:19.313049821 +0200
@@ -42,9 +42,8 @@
 #include <mach/mach_port.h>
 #include <sys/sysctl.h>
 
-
-#elif defined (__linux__)
-
+#elif defined (__linux__) || defined(__FreeBSD_kernel__)
+/*  FreeBSD kernel with glibc */
 #include <sys/wait.h>
 
 #elif defined (__FreeBSD__)
